// react-wallets/dist/index.js
function _interopDefault (ex) { return (ex && (typeof ex === 'object') && 'default' in ex) ? ex['default'] : ex; }

var React = require('react');
var React__default = _interopDefault(React);
var styled = _interopDefault(require('styled-components'));

function _taggedTemplateLiteralLoose(strings, raw) {
  if (!raw) {
    raw = strings.slice(0);
  }
  strings.raw = raw;
  return strings;
}

var _templateObject, _templateObject2, _templateObject3, _templateObject4;
var Container = styled("div")(_templateObject || (_templateObject = _taggedTemplateLiteralLoose(["\n  display: flex;\n  width: 100%;\n  flex-direction: column;\n  -webkit-box-align: center;\n  align-items: center;\n  margin-top: ", ";\n  margin-bottom: ", ";\n"])), function (_ref) {
  var type = _ref.type;
  return type === "code128" || type === "pdf417" ? "90px" : "100px";
}, function (_ref2) {
  var type = _ref2.type;
  return type === "code128" || type === "pdf417" ? "30px" : "45px";
});
var ImageContainer = styled("div")(_templateObject2 || (_templateObject2 = _taggedTemplateLiteralLoose(["\n  display: flex;\n  flex-direction: column;\n  justify-content: center;\n  align-items: center;\n  margin: auto;\n  padding: 10px;\n  border-radius: 4px;\n  box-sizing: border-box;\n  background: #fff;\n"])));
var Image = styled("img")(_templateObject3 || (_templateObject3 = _taggedTemplateLiteralLoose(["\n  height: ", ";\n  width: ", ";\n  object-fit: cover;\n"])), function (_ref3) {
  var type = _ref3.type;
  return type === "aztec" || type === "qrCode" ? "128px" : "45px";
}, function (_ref4) {
  var type = _ref4.type;
  return type === "aztec" || type === "qrCode" ? "128px" : "225px";
});
var AccessCode = styled("p")(_templateObject4 || (_templateObject4 = _taggedTemplateLiteralLoose(["\n  text-align: center;\n  font-family: \"Product Sans\", sans-serif;\n  color: #000;\n  margin: ", ";\n  font-size: 14px;\n  font-weight: 450;\n"])), function (_ref5) {
  var type = _ref5.type;
  return type === "aztec" || type === "qrCode" ? "3px auto" : "5px 0px";
});

var Access = function Access(props) {
  var accessType = props.accessType,
    accessImage = props.accessImage,
    accessCodeId = props.accessCodeId;
  return React__default.createElement(Container, {
    type: accessType
  }, React__default.createElement(ImageContainer, null, React__default.createElement(Image, {
    src: accessImage,
    type: accessType,
    alt: accessType
  }), accessCodeId && React__default.createElement(AccessCode, {
    type: accessType
  }, accessCodeId)));
};

var _templateObject$1, _templateObject2$1;
var BannerContainer = styled("div")(_templateObject$1 || (_templateObject$1 = _taggedTemplateLiteralLoose(["\n  height: 125px;\n  width: 100%;\n  overflow: hidden;\n  box-sizing: border-box;\n"])));
var BannerImage = styled("img")(_templateObject2$1 || (_templateObject2$1 = _taggedTemplateLiteralLoose(["\n  object-fit: cover;\n  width: 100%;\n  height: inherit;\n"])));

var Banner = function Banner(props) {
  var bannerImage = props.bannerImage;
  if (bannerImage) {
    return React__default.createElement(BannerContainer, null, React__default.createElement(BannerImage, {
      src: bannerImage,
      alt: 'banner-image'
    }));
  } else return null;
};

var _templateObject$2, _templateObject2$2, _templateObject3$1, _templateObject4$1, _templateObject5, _templateObject6, _templateObject7;
var HeaderContainer = styled("div")(_templateObject$2 || (_templateObject$2 = _taggedTemplateLiteralLoose(["\n  display: flex;\n  align-items: center;\n  justify-content: space-between;\n  padding: 6px 12px 2px 12px;\n"])));
var LogoContainer = styled("img")(_templateObject2$2 || (_templateObject2$2 = _taggedTemplateLiteralLoose(["\n  display: flex;\n  height: 40px;\n  border: 1px solid #a0b5ba;\n  border-radius: 50%;\n  img {\n    height: 32px;\n    width: 32px;\n    object-fit: contain;\n  }\n"])));
var LeftContainer = styled("div")(_templateObject3$1 || (_templateObject3$1 = _taggedTemplateLiteralLoose(["\n  display: flex;\n  align-items: center;\n  gap: 15px;\n"])));
var RightContainer = styled("div")(_templateObject4$1 || (_templateObject4$1 = _taggedTemplateLiteralLoose(["\n  display: flex;\n  flex-direction: column;\n  margin-bottom: 15px;\n"])));
var HeaderTitle = styled("h4")(_templateObject5 || (_templateObject5 = _taggedTemplateLiteralLoose(["\n  font-size: 1.1rem;\n  font-weight: 450;\n  color: #000;\n  text-align: center;\n"])));
var HeaderSubTitle = styled("h3")(_templateObject6 || (_templateObject6 = _taggedTemplateLiteralLoose(["\n  color: ", ";\n  font-size: 0.6rem;\n  font-weight: 650;\n  margin: 0px;\n  text-align: end;\n"])), function (_ref) {
  var color = _ref.color;
  return color || "#000";
});
var HeaderHeading = styled("h3")(_templateObject7 || (_templateObject7 = _taggedTemplateLiteralLoose(["\n  color: \"#000\";\n  font-size: 1rem;\n  font-weight: 400;\n  margin: 0px;\n  text-align: end;\n"])));

var Header = function Header(props) {
  var headerLogoLink = props.headerLogoLink,
    headerTitle = props.headerTitle,
    title = props.title,
    mainTitle = props.mainTitle;
  return React__default.createElement(HeaderContainer, null, React__default.createElement(LeftContainer, null, React__default.createElement(LogoContainer, {
    src: headerLogoLink
  }), React__default.createElement(HeaderTitle, null, headerTitle)), React__default.createElement(RightContainer, null, React__default.createElement(HeaderSubTitle, {
    color: '#446A6B'
  }, title), React__default.createElement(HeaderHeading, null, mainTitle)));
};

function MoreHorizIcon(_ref) {
  var color = _ref.color;
  return React__default.createElement("svg", {
    xmlns: 'http://www.w3.org/2000/svg',
    width: '24',
    height: '24'
  }, React__default.createElement("path", {
    fill: color,
    d: 'M6 14q-.825 0-1.412-.588Q4 12.825 4 12t.588-1.413Q5.175 10 6 10t1.412.587Q8 11.175 8 12q0 .825-.588 1.412Q6.825 14 6 14zm6 0q-.825 0-1.412-.588Q10 12.825 10 12t.588-1.413Q11.175 10 12 10t1.413.587Q14 11.175 14 12q0 .825-.587 1.412Q12.825 14 12 14zm6 0q-.825 0-1.413-.588Q16 12.825 16 12t.587-1.413Q17.175 10 18 10q.825 0 1.413.587Q20 11.175 20 12q0 .825-.587 1.412Q18.825 14 18 14z'
  }));
}

var PlaneIcon = function PlaneIcon() {
  return React__default.createElement("svg", {
    width: '20',
    height: '20',
    viewBox: '0 0 20 19',
    xmlns: 'http://www.w3.org/2000/svg'
  }, React__default.createElement("path", {
    d: 'M8,21h2l5-8h5.5a1.5,1.5,0,0,0,0-3H15L10,2H8l2.5,8H5L3.5,8H2l1,3.5L2,15H3.5L5,13h5.5Z',
    transform: 'translate(-2 -2)'
  }));
};

function ArrowIcon() {
  return React__default.createElement("svg", {
    xmlns: 'http://www.w3.org/2000/svg',
    width: '16',
    height: '16',
    fill: 'none',
    viewBox: '0 0 25 25'
  }, React__default.createElement("path", {
    stroke: '#000',
    strokeLinecap: 'round',
    strokeLinejoin: 'round',
    strokeWidth: '2',
    d: 'M12.5 1L24 12.5 12.5 24'
  }), React__default.createElement("path", {
    fill: '#000',
    d: 'M1 11.5a1 1 0 100 2v-2zm23 0H1v2h23v-2z'
  }));
}

var EmailIcon = function EmailIcon(_ref) {
  var _ref$fill = _ref.fill,
    fill = _ref$fill === void 0 ? '#404040' : _ref$fill,
    _ref$height = _ref.height,
    height = _ref$height === void 0 ? 10 : _ref$height,
    _ref$width = _ref.width,
    width = _ref$width === void 0 ? 13 : _ref$width;
  return React__default.createElement("svg", {
    xmlns: 'http://www.w3.org/2000/svg',
    width: width,
    height: height,
    fill: 'none',
    viewBox: '0 0 13 10'
  }, React__default.createElement("path", {
    fill: fill,
    fillRule: 'evenodd',
    d: 'M.93.044a1.232 1.232 0 00-.891 1.48c.05.196.143.372.271.508.114.121 5.522 3.828 5.72 3.92.301.14.661.135.969-.014.241-.117 5.54-3.752 5.674-3.892.632-.663.287-1.785-.616-2.004C11.82-.015 1.162-.014.93.044zM0 5.84C0 7.95.009 8.916.028 9.012c.093.459.455.833.914.944.239.058 10.833.06 11.1.002.461-.1.835-.48.929-.946.032-.159.04-6.21.009-6.195-.01.005-1.238.843-2.728 1.863C8.762 5.699 7.46 6.575 7.36 6.627c-.281.145-.519.2-.862.2-.342 0-.58-.055-.86-.2-.1-.052-1.401-.928-2.892-1.947A769.266 769.266 0 00.02 2.817C.009 2.813 0 4.173 0 5.841z',
    clipRule: 'evenodd'
  }));
};

var PhoneIcon = function PhoneIcon(_ref) {
  var _ref$fill = _ref.fill,
    fill = _ref$fill === void 0 ? '#404040' : _ref$fill,
    _ref$height = _ref.height,
    height = _ref$height === void 0 ? 13 : _ref$height,
    _ref$width = _ref.width,
    width = _ref$width === void 0 ? 13 : _ref$width;
  return React__default.createElement("svg", {
    xmlns: 'http://www.w3.org/2000/svg',
    width: width,
    height: height,
    fill: 'none',
    viewBox: '0 0 13 13'
  }, React__default.createElement("path", {
    fill: fill,
    fillRule: 'evenodd',
    d: 'M2.25.034c-.315.069-.376.116-1.2.938-.718.716-.774.778-.858.957-.139.294-.174.453-.189.869-.015.42.02.697.147 1.18.765 2.907 4.198 6.73 7.483 8.336.975.476 1.71.684 2.436.686.444.001.668-.043.976-.195.198-.097.25-.143.966-.856.8-.798.849-.858.95-1.196a1.52 1.52 0 000-.69c-.092-.307-.13-.351-1.142-1.367-.535-.537-1.041-1.024-1.125-1.084-.506-.359-1.156-.31-1.58.121-.127.13-.177.207-.265.411a2.38 2.38 0 01-.178.345c-.243.33-.714.539-1.114.496-.597-.063-1.487-.6-2.241-1.353-.913-.91-1.452-1.959-1.32-2.57a1.27 1.27 0 01.498-.74c.058-.038.18-.101.27-.14.266-.113.33-.15.454-.257.424-.37.517-1.06.208-1.551C5.335 2.229 3.576.456 3.362.292a1.47 1.47 0 00-.597-.268 1.159 1.159 0 00-.515.01z',
    clipRule: 'evenodd'
  }));
};

var PositionIcon = function PositionIcon(_ref) {
  var _ref$fill = _ref.fill,
    fill = _ref$fill === void 0 ? '#404040' : _ref$fill,
    _ref$height = _ref.height,
    height = _ref$height === void 0 ? 14 : _ref$height,
    _ref$width = _ref.width,
    width = _ref$width === void 0 ? 10 : _ref$width;
  return React__default.createElement("svg", {
    xmlns: 'http://www.w3.org/2000/svg',
    width: width,
    height: height,
    fill: 'none',
    viewBox: '0 0 10 14'
  }, React__default.createElement("path", {
    fill: fill,
    fillRule: 'evenodd',
    d: 'M4.253.016C2.448.232.965 1.372.309 3.048a4.657 4.657 0 00.082 3.54c.155.354 3.67 6.616 3.787 6.748.058.064.174.149.26.19a.66.66 0 00.36.074c.265-.001.449-.077.608-.253.127-.139 3.626-6.362 3.798-6.754a4.672 4.672 0 00.358-2.456A4.768 4.768 0 005.603.053a7.422 7.422 0 00-1.35-.037zm1.132 2.405c.31.096.54.236.792.482.249.243.392.466.496.772.064.187.072.254.072.572 0 .317-.008.384-.072.572a1.825 1.825 0 01-.496.772c-.25.243-.478.383-.792.484-.193.062-.261.07-.587.07-.325 0-.394-.008-.586-.07-.312-.1-.54-.24-.792-.484a1.816 1.816 0 01-.497-.772c-.063-.188-.072-.255-.072-.572 0-.314.009-.385.07-.564.107-.317.25-.543.497-.783a1.941 1.941 0 011.04-.543c.239-.042.683-.012.927.064z',
    clipRule: 'evenodd'
  }));
};

var ExpandArrowIcon = function ExpandArrowIcon() {
  return React__default.createElement("svg", {
    xmlns: 'http://www.w3.org/2000/svg',
    width: '9',
    height: '14',
    fill: 'none',
    viewBox: '0 0 9 14'
  }, React__default.createElement("path", {
    stroke: '#1C2843',
    strokeLinecap: 'round',
    strokeLinejoin: 'round',
    strokeWidth: '2',
    d: 'M7.223 1l-6 6 6 6'
  }));
};

var GlobeIcon = function GlobeIcon(_ref) {
  var _ref$fill = _ref.fill,
    fill = _ref$fill === void 0 ? '#404040' : _ref$fill,
    _ref$height = _ref.height,
    height = _ref$height === void 0 ? 49 : _ref$height,
    _ref$width = _ref.width,
    width = _ref$width === void 0 ? 49 : _ref$width;
  return React__default.createElement("svg", {
    xmlns: 'http://www.w3.org/2000/svg',
    width: width,
    height: height,
    fill: 'none',
    viewBox: '0 0 49 49'
  }, React__default.createElement("g", {
    clipPath: 'url(#clip0_1484_42025)'
  }, React__default.createElement("path", {
    fill: fill,
    fillRule: 'evenodd',
    d: 'M24.644.029c1.205.019 2.545.086 2.966.153.421.057 1.302.22 1.962.354.66.134 1.819.45 2.584.708.766.249 1.886.68 2.489.948.602.277 1.655.832 2.335 1.234.67.402 1.665 1.062 2.2 1.464a31.49 31.49 0 011.81 1.513c.459.42 1.186 1.148 1.607 1.607.431.47 1.11 1.283 1.513 1.81a31.384 31.384 0 011.464 2.2c.402.68.957 1.732 1.234 2.335.268.603.699 1.723.948 2.489.258.765.574 1.923.708 2.584.134.66.306 1.722.383 2.373.086.641.153 1.838.153 2.651 0 .814-.067 2.03-.144 2.709-.086.67-.258 1.75-.392 2.411-.134.66-.45 1.819-.708 2.584-.249.766-.68 1.886-.948 2.489a31.273 31.273 0 01-1.234 2.335 31.397 31.397 0 01-1.464 2.2 31.575 31.575 0 01-1.513 1.81 31.87 31.87 0 01-1.607 1.607c-.47.431-1.283 1.11-1.81 1.513a31.397 31.397 0 01-2.2 1.464c-.68.402-1.732.957-2.335 1.234-.603.268-1.723.699-2.489.948-.765.258-1.923.574-2.584.708-.66.134-1.588.297-2.057.364-.498.076-1.761.124-3.015.124-1.254 0-2.517-.048-3.015-.124a42.187 42.187 0 01-2.057-.364c-.66-.134-1.819-.45-2.584-.708a31.145 31.145 0 01-2.489-.948 31.273 31.273 0 01-2.335-1.234 31.387 31.387 0 01-2.2-1.464 31.534 31.534 0 01-1.81-1.513 31.877 31.877 0 01-1.607-1.607 31.53 31.53 0 01-1.513-1.81 31.397 31.397 0 01-1.464-2.2 31.306 31.306 0 01-1.234-2.335 31.142 31.142 0 01-.948-2.489 25.954 25.954 0 01-.708-2.584A34.938 34.938 0 01.153 27.2C.067 26.558 0 25.36 0 24.548c0-.814.067-2.029.144-2.708.086-.67.258-1.752.392-2.412.134-.66.45-1.819.708-2.584.249-.766.68-1.886.948-2.489.277-.603.832-1.655 1.234-2.335a31.384 31.384 0 011.464-2.2 31.489 31.489 0 011.513-1.81A31.844 31.844 0 018.01 6.403c.47-.431 1.283-1.11 1.81-1.513a31.387 31.387 0 012.2-1.464c.68-.402 1.732-.957 2.335-1.234a31.13 31.13 0 012.489-.948 25.945 25.945 0 012.584-.708c.66-.134 1.579-.287 2.057-.345.47-.057.88-.134.91-.153.019-.028 1.033-.028 2.249-.01zm-2.537 3.33c-.784.086-1.11.153-1.129.24-.01.076.249.813.565 1.646.325.823.689 1.799.823 2.172.124.364.297.957.373 1.32.077.355.134.92.134 1.264 0 .345-.048.852-.105 1.13-.058.267-.21.717-.335.976a3.23 3.23 0 01-.584.842c-.201.21-.555.469-.794.593-.24.115-.565.24-.718.277-.163.039-.718.134-1.244.211-.91.134-.986.163-1.627.632-.373.268-.91.603-1.197.737-.497.23-.612.239-2.172.249-.9 0-1.675.028-1.723.076-.038.038-.038.105.02.163.047.048.354.239.67.43.315.192.918.45 1.34.584.42.134 1.32.326 2.01.421.679.105 1.482.268 1.77.364.287.105.65.249.813.335.153.076.498.335.756.555.259.23.632.66.814.947.191.288.507.929.708 1.417.191.478.392.88.45.88.047 0 .363-.162.698-.363.326-.21.766-.431.976-.498.211-.067.709-.124 1.101-.124.479.01.919.076 1.34.22.344.114.861.354 1.148.526.287.182.766.536 1.053.794.287.25.699.718.91 1.034.21.306.487.89.612 1.282.191.584.22.852.181 1.436-.028.421-.134.91-.258 1.196a5.979 5.979 0 01-.421.795c-.115.172-1.005.995-1.981 1.828-.976.832-1.952 1.713-2.182 1.961-.22.24-.402.47-.402.508 0 .047.067.44.144.89.076.45.143.966.143 1.148 0 .182-.067.565-.143.833-.096.335-.288.66-.546.957-.23.239-.622.545-.89.66a4.039 4.039 0 01-.766.278.995.995 0 00-.402.162c-.067.067-.181.622-.267 1.225-.077.603-.211 1.34-.297 1.627a6.87 6.87 0 01-.335.88c-.096.202-.345.537-.536.757-.201.22-.498.469-.65.565-.164.105-.546.248-.862.334-.46.115-.68.125-1.1.048a4.59 4.59 0 01-.948-.316c-.23-.114-.594-.401-.814-.64-.23-.24-.517-.651-.65-.91a9.621 9.621 0 01-.355-.813c-.057-.182-.172-.823-.249-1.417-.076-.584-.21-1.215-.306-1.388a13.64 13.64 0 00-.775-1.167c-.344-.479-.766-1.168-.938-1.531a6.287 6.287 0 01-.411-1.245c-.048-.315-.067-1.11-.039-1.77.048-.995.03-1.254-.105-1.512-.086-.163-.565-.766-1.072-1.34-.507-.565-1.033-1.244-1.186-1.503a6.966 6.966 0 01-.45-1.052c-.134-.46-.163-.842-.144-1.895l.02-1.311c-1.15-.852-1.81-1.369-2.202-1.713a36.835 36.835 0 01-1.512-1.388c-.44-.43-.976-1.005-1.196-1.282-.22-.278-.43-.47-.46-.431-.038.038-.153.402-.258.794a24.209 24.209 0 00-.593 3.11c-.058.47-.115 1.59-.115 2.489 0 .89.048 1.99.105 2.44.067.45.201 1.264.306 1.819.115.555.412 1.646.68 2.44.258.785.708 1.905.986 2.488a29.17 29.17 0 001.052 1.914c.297.47.842 1.273 1.216 1.77.373.499 1.225 1.456 1.885 2.116.66.66 1.617 1.512 2.115 1.885.498.374 1.292.92 1.77 1.216a22.052 22.052 0 004.068 1.942c.603.211 1.55.489 2.105.623.555.134 1.522.315 2.154.411.708.096 1.79.153 2.823.153.919 0 2.038-.048 2.488-.105.45-.067 1.264-.201 1.819-.316a22.544 22.544 0 002.153-.565 24.62 24.62 0 002.153-.813 21.81 21.81 0 001.77-.88c.422-.24 1.092-.651 1.484-.92.393-.258 1.13-.813 1.627-1.215.498-.411 1.149-.976 1.436-1.263l.526-.526c-1.235-1.197-1.54-1.589-1.76-2.03-.164-.344-.355-.899-.412-1.243-.087-.479-.087-.814 0-1.436.057-.45.287-1.311.497-1.914.22-.603.46-1.426.536-1.818.077-.393.115-.89.086-1.1a1.198 1.198 0 00-.316-.642c-.143-.144-.67-.412-1.167-.603a20.632 20.632 0 01-1.914-.88 23.55 23.55 0 01-1.77-1.034 8.432 8.432 0 01-1.178-.919c-.23-.23-.526-.66-.66-.947-.182-.364-.268-.737-.297-1.197-.028-.373 0-.947.067-1.292.067-.344.201-.9.316-1.244.105-.344.411-1.052.689-1.579.345-.67.68-1.158 1.12-1.627.354-.363.909-.832 1.244-1.033.335-.201.91-.45 1.282-.565.536-.163.842-.191 1.532-.153.727.038 1.12.134 2.536.612 1.665.565 1.684.565 2.794.565.613 0 1.14-.02 1.177-.048.03-.029-.057-.325-.191-.67a28.098 28.098 0 00-.737-1.579 24.362 24.362 0 00-1.187-2.01 25.542 25.542 0 00-.928-1.34l-.24-.287c-.344.326-.66.517-.928.642-.373.172-.65.22-1.244.22a4.975 4.975 0 01-1.579-.268c-.45-.134-.909-.259-1.033-.259-.115 0-.67.201-1.235.45-.88.373-1.148.45-1.8.488-.63.029-.87 0-1.339-.182-.46-.172-.737-.373-1.35-.985-.516-.527-.86-.977-1.071-1.398-.24-.469-.306-.746-.306-1.148 0-.287.047-.68.105-.861.067-.182.249-.527.411-.766.153-.24.737-.89 1.302-1.464.555-.565.986-1.043.957-1.063-.029-.019-.46-.162-.957-.325a23.54 23.54 0 00-1.77-.488 35.551 35.551 0 00-1.915-.335 25.928 25.928 0 00-2.01-.172c-.526-.02-1.215-.02-1.53-.01-.317.01-1.073.077-1.676.134z',
    clipRule: 'evenodd'
  })), React__default.createElement("defs", null, React__default.createElement("clipPath", {
    id: 'clip0_1484_42025'
  }, React__default.createElement("path", {
    fill: '#fff',
    d: 'M0 0H49V49H0z'
  }))));
};

function Plane() {
  return React__default.createElement("svg", {
    width: '33',
    height: '33',
    viewBox: '0 0 141 136'
  }, React__default.createElement("path", {
    d: 'M124 61h-20L88 42s8-1 8-4c0-4-12-4-12-4L72 21s9 0 9-3c0-4-12-3-12-3L55 0h-9l19 48s2.8 5.8 0 10c-2 3-40 5-40 5L10 44H0l7 21s-4 1-4 3 4 3 4 3L0 92h10l15-19s38 2 40 5c2.8 4.2 0 10 0 10l-19 48h9l14-15s12 1 12-3c0-3-9-3-9-3l12-13s12 0 12-4c0-3-8-4-8-4l16-19h20c13 0 17-4 17-7s-4-7-17-7z'
  }));
}

function Boat() {
  return React__default.createElement("svg", {
    xmlns: 'http://www.w3.org/2000/svg',
    width: '33',
    height: '32',
    fill: 'none',
    viewBox: '0 0 33 32'
  }, React__default.createElement("path", {
    fill: '#000',
    fillRule: 'evenodd',
    d: 'M11.322.32c-.365.208-.533.671-.533 1.473 0 1.14-.027 1.17-1.066 1.17-1.015 0-1.066.05-1.066 1.037 0 .94-.08 1.037-.856 1.037h-.857l-.972 6.147c-.534 3.38-.97 6.302-.97 6.493 0 .19 2.563-1.176 5.695-3.037 3.132-1.86 5.778-3.382 5.879-3.382.1 0 2.767 1.535 5.925 3.41 3.158 1.876 5.786 3.367 5.84 3.315.054-.053-.239-2.888-.652-6.299-.413-3.411-.752-6.303-.755-6.425-.002-.122-.346-.222-.765-.222-.667 0-.761-.128-.761-1.037 0-.987-.051-1.037-1.066-1.037-.963 0-1.066-.08-1.066-.822 0-.453-.214-1.12-.474-1.482-.95-1.318-2.572-.498-2.572 1.299v1.025L16.956 2.9l-3.273-.084-.101-1.035c-.163-1.664-.974-2.188-2.26-1.46zm1.283 7.22c.201.237.319.67.261.963-.089.448-.421.533-2.077.533-1.655 0-1.988-.085-2.076-.533-.182-.916.528-1.393 2.076-1.393.895 0 1.589.165 1.816.43zm5.786 0c.202.237.32.67.262.963-.09.448-.422.533-2.077.533s-1.988-.085-2.077-.533c-.181-.916.528-1.393 2.077-1.393.894 0 1.588.165 1.815.43zm5.786 0c.202.237.32.67.262.963-.089.448-.422.533-2.077.533s-1.988-.085-2.076-.533c-.182-.916.528-1.393 2.076-1.393.895 0 1.589.165 1.815.43zM7.767 18.016C3.506 20.571.011 22.744 0 22.843c-.012.1.925 2.201 2.082 4.67L4.186 32l12.782-.077 12.782-.077 1.759-4.444c1.33-3.361 1.673-4.527 1.407-4.786-.49-.478-15.468-9.343-15.588-9.227-.054.053-.196 2.994-.317 6.536-.122 3.542-.337 6.44-.48 6.44-.143 0-.26-1.032-.261-2.295-.002-3.698-.382-10.74-.58-10.72-.098.01-3.665 2.11-7.924 4.665z',
    clipRule: 'evenodd'
  }));
}

function Bus() {
  return React__default.createElement("svg", {
    xmlns: 'http://www.w3.org/2000/svg',
    width: '26',
    height: '31',
    fill: 'none',
    viewBox: '0 0 26 31'
  }, React__default.createElement("path", {
    fill: '#000',
    fillRule: 'evenodd',
    d: 'M4.865.09C3.437.323 2.252.91 1.396 1.81.805 2.43.41 3.164.156 4.113L0 4.7v20.605l.135.342c.263.679.75 1.08 1.465 1.21l.351.064.026 1.252c.015.808.043 1.25.142 1.574.078.256.203.438.402.667.387.445.735.583 1.465.585.526.002.621-.015.92-.167.389-.197.655-.488.85-.93.197-.446.252-.877.252-2.006v-.986h13.951l.002.986c0 1.129.055 1.56.252 2.007.195.441.46.732.85.93.298.15.396.168.92.167.663 0 1.015-.123 1.367-.471.515-.511.652-1.045.654-2.538v-1.072l.338-.071c.596-.125 1.043-.406 1.318-.821.131-.198.223-.425.274-.68.06-.3.08-9.756.056-15.772-.01-2.652-.027-4.635-.054-4.874-.172-1.554-.778-2.841-1.682-3.577-.68-.551-1.455-.883-2.408-1.033-.778-.12-16.237-.12-16.98 0zM5 3.5A1.5 1.5 0 016.5 2h13a1.5 1.5 0 010 3h-13A1.5 1.5 0 015 3.5zM5 8a2 2 0 00-2 2v4a2 2 0 002 2h16a2 2 0 002-2v-4a2 2 0 00-2-2H5zm18 13a2 2 0 11-4 0 2 2 0 014 0zM5 23a2 2 0 100-4 2 2 0 000 4z',
    clipRule: 'evenodd'
  }));
}

function Train() {
  return React__default.createElement("svg", {
    xmlns: 'http://www.w3.org/2000/svg',
    width: '20',
    height: '30',
    fill: 'none',
    viewBox: '0 0 20 30'
  }, React__default.createElement("path", {
    fill: '#000',
    fillRule: 'evenodd',
    d: 'M6 0a5 5 0 00-5 5v10a5.002 5.002 0 003.783 4.85l-1.549 2.5-1.6 2.712C.882 26.343.47 27.055.249 27.483c-.078.149-.13.264-.168.356a.947.947 0 00-.074.267c-.035.37.09.7.318.922.184.18.434.287.72.287.6 0 .763-.15 1.474-1.365l.54-.922L4.257 25h11.322l1.197 2.027.51.875c.74 1.266.895 1.412 1.504 1.412.639 0 1.1-.539 1.04-1.21-.028-.279-.278-.748-1.63-3.039l-1.597-2.709-1.403-2.5A5.002 5.002 0 0019 15V5A4.982 4.982 0 0016.553.7 4.974 4.974 0 0014 0H6zm6.768 20H7.203l-1.457 2.459L5.43 23h8.974l-.318-.541L12.768 20zM14 1H6v3h8V1zM3 7a2 2 0 012-2h10a2 2 0 012 2v4a1.919 1.919 0 01-.084.577A1.995 1.995 0 0115 13H5a2 2 0 01-2-2V7zm1.5 11A1.497 1.497 0 006 16.5 1.5 1.5 0 104.5 18zM17 16.5a1.5 1.5 0 11-3 0 1.5 1.5 0 013 0z',
    clipRule: 'evenodd'
  }));
}

function MoreIcon(_ref) {
  var color = _ref.color;
  return React__default.createElement("svg", {
    xmlns: 'http://www.w3.org/2000/svg',
    width: '24',
    height: '24'
  }, React__default.createElement("path", {
    fill: color,
    d: 'M6 14q-.825 0-1.412-.588Q4 12.825 4 12t.588-1.413Q5.175 10 6 10t1.412.587Q8 11.175 8 12q0 .825-.588 1.412Q6.825 14 6 14zm6 0q-.825 0-1.412-.588Q10 12.825 10 12t.588-1.413Q11.175 10 12 10t1.413.587Q14 11.175 14 12q0 .825-.587 1.412Q12.825 14 12 14zm6 0q-.825 0-1.413-.588Q16 12.825 16 12t.587-1.413Q17.175 10 18 10q.825 0 1.413.587Q20 11.175 20 12q0 .825-.587 1.412Q18.825 14 18 14z'
  }));
}

var Svgs = {
  MoreHorizIcon: MoreHorizIcon,
  PlaneIcon: PlaneIcon,
  ArrowIcon: ArrowIcon,
  ExpandArrowIcon: ExpandArrowIcon,
  Plane: Plane,
  Boat: Boat,
  Bus: Bus,
  Train: Train,
  MoreIcon: MoreIcon,
  EmailIcon: EmailIcon,
  PhoneIcon: PhoneIcon,
  PositionIcon: PositionIcon,
  GlobeIcon: GlobeIcon
};

var _templateObject$3, _templateObject2$3, _templateObject3$2, _templateObject4$2, _templateObject5$1, _templateObject6$1, _templateObject7$1;
var InformationWrapper = styled("div")(_templateObject$3 || (_templateObject$3 = _taggedTemplateLiteralLoose(["\n  display: flex;\n  flex-direction: column;\n  gap: 20px;\n  margin: 10px 0px;\n  padding: 0px 12px;\n"])));
var InformationsTitleWrapper = styled("div")(_templateObject2$3 || (_templateObject2$3 = _taggedTemplateLiteralLoose(["\n  display: flex;\n  align-items: center;\n"])));
var InformationContainer = styled("div")(_templateObject3$2 || (_templateObject3$2 = _taggedTemplateLiteralLoose(["\n  display: flex;\n  flex-direction: column;\n  flex: 1;\n  text-align: ", ";\n"])), function (_ref) {
  var textAlign = _ref.textAlign;
  return textAlign || "left";
});
var InformationTitle = styled("h3")(_templateObject4$2 || (_templateObject4$2 = _taggedTemplateLiteralLoose(["\n  color: #446a6b;\n  font-size: 0.7rem;\n  font-weight: 650;\n  margin: 0px;\n  text-transform: uppercase;\n"])));
var InformationValue = styled("h3")(_templateObject5$1 || (_templateObject5$1 = _taggedTemplateLiteralLoose(["\n  color: #000;\n  font-size: 1rem;\n  font-weight: 400;\n  margin: 0px;\n  padding: 0px;\n  white-space: nowrap;\n"])));
var FromToTitle = styled("h3")(_templateObject6$1 || (_templateObject6$1 = _taggedTemplateLiteralLoose(["\n  width: 100%;\n  min-height: 11px;\n  font-size: 0.7rem;\n  font-weight: 500;\n  margin: 0px;\n  text-transform: uppercase;\n  white-space: nowrap;\n"])));
var FromToValue = styled("h3")(_templateObject7$1 || (_templateObject7$1 = _taggedTemplateLiteralLoose(["\n  font-size: 1.9rem;\n  font-weight: 400;\n  margin: 0px;\n  color: rgb(0, 52, 69);\n  white-space: nowrap;\n  overflow: hidden;\n  text-overflow: ellipsis;\n  text-transform: ", ";\n"])), function (_ref2) {
  var textTransform = _ref2.textTransform;
  return textTransform || "none";
});

var Information = function Information(props) {
  var type = props.type,
    LeftTitle = props.LeftTitle,
    LeftValue = props.LeftValue,
    RightTitle = props.RightTitle,
    RightValue = props.RightValue,
    topLeftTitle = props.topLeftTitle,
    topLeftValue = props.topLeftValue,
    topCenterLeftTitle = props.topCenterLeftTitle,
    topCenterLeftValue = props.topCenterLeftValue,
    topCenterRightTitle = props.topCenterRightTitle,
    topCenterRightValue = props.topCenterRightValue,
    topRightTitle = props.topRightTitle,
    topRightValue = props.topRightValue,
    bottomLeftTitle = props.bottomLeftTitle,
    bottomLeftValue = props.bottomLeftValue,
    bottomCenterTitle = props.bottomCenterTitle,
    bottomCenterValue = props.bottomCenterValue,
    bottomRightTitle = props.bottomRightTitle,
    bottomRightValue = props.bottomRightValue;
  var renderSvgByType = function renderSvgByType(pass) {
    if (pass === 'boat') {
      return React__default.createElement(Svgs.Boat, null);
    }
    if (pass === 'plane') {
      return React__default.createElement(Svgs.Plane, null);
    }
    if (pass === 'bus') {
      return React__default.createElement(Svgs.Bus, null);
    }
    if (pass === 'train' || pass === 'subway') {
      return React__default.createElement(Svgs.Train, null);
    } else return null;
  };
  return React__default.createElement(InformationWrapper, null, LeftTitle && LeftValue && RightTitle && RightValue && React__default.createElement(InformationsTitleWrapper, null, React__default.createElement(InformationContainer, null, React__default.createElement(FromToTitle, null, LeftTitle), React__default.createElement(FromToValue, {
    textTransform: 'uppercase'
  }, LeftValue)), renderSvgByType(type), React__default.createElement(InformationContainer, {
    textAlign: 'right'
  }, React__default.createElement(FromToTitle, null, RightTitle), React__default.createElement(FromToValue, {
    textTransform: 'uppercase'
  }, RightValue))), React__default.createElement(InformationsTitleWrapper, null, topLeftTitle && topLeftValue && React__default.createElement(InformationContainer, null, React__default.createElement(InformationTitle, null, topLeftTitle), React__default.createElement(InformationValue, null, topLeftValue)), topCenterLeftTitle && topCenterLeftValue && React__default.createElement(InformationContainer, null, React__default.createElement(InformationTitle, null, topCenterLeftTitle), React__default.createElement(InformationValue, null, topCenterLeftValue)), topCenterRightTitle && topCenterRightValue && React__default.createElement(InformationContainer, null, React__default.createElement(InformationTitle, null, topCenterRightTitle), React__default.createElement(InformationValue, null, topCenterRightValue)), topRightTitle && topRightValue && React__default.createElement(InformationContainer, {
    textAlign: 'right'
  }, React__default.createElement(InformationTitle, null, topRightTitle), React__default.createElement(InformationValue, null, topRightValue))), React__default.createElement(InformationsTitleWrapper, null, bottomLeftTitle && bottomLeftValue && React__default.createElement(InformationContainer, null, React__default.createElement(InformationTitle, null, bottomLeftTitle), React__default.createElement(InformationValue, null, bottomLeftValue)), bottomCenterTitle && bottomCenterValue && React__default.createElement(InformationContainer, {
    textAlign: 'center'
  }, React__default.createElement(InformationTitle, null, bottomCenterTitle), React__default.createElement(InformationValue, null, bottomCenterValue)), bottomRightTitle && bottomRightValue && React__default.createElement(InformationContainer, {
    textAlign: 'right'
  }, React__default.createElement(InformationTitle, null, bottomRightTitle), React__default.createElement(InformationValue, null, bottomRightValue))));
};

var _templateObject$4, _templateObject2$4, _templateObject3$3, _templateObject4$3, _templateObject5$2, _templateObject6$2;
var Container$1 = styled("div")(_templateObject$4 || (_templateObject$4 = _taggedTemplateLiteralLoose(["\n  display: flex;\n  flex-direction: column;\n  gap: 5px;\n  padding: 8px 12px 2px 12px;\n"])));
var Section = styled("div")(_templateObject2$4 || (_templateObject2$4 = _taggedTemplateLiteralLoose(["\n  display: flex;\n  flex-direction: column;\n  align-items: flex-start;\n"])));
var Row = styled("div")(_templateObject3$3 || (_templateObject3$3 = _taggedTemplateLiteralLoose(["\n  display: flex;\n  gap: 3px;\n"])));
var Title = styled("h3")(_templateObject4$3 || (_templateObject4$3 = _taggedTemplateLiteralLoose(["\n  color: #939393;\n  font-weight: 450;\n  font-size: 0.72rem;\n  letter-spacing: 0.025em;\n  margin: 5px 0px;\n  text-transform: ", ";\n"])), function (_ref) {
  var textTransform = _ref.textTransform;
  return textTransform || "uppercase";
});
var Value = styled("h3")(_templateObject5$2 || (_templateObject5$2 = _taggedTemplateLiteralLoose(["\n  color: ", ";\n  max-width: 350px;\n  font-weight: 450;\n  font-size: 0.92rem;\n  margin: 3px 0px;\n  word-break: break-all;\n  overflow-wrap: break-word;\n"])), function (_ref2) {
  var color = _ref2.color;
  return color || "#939393";
});
var Divider = styled("span")(_templateObject6$2 || (_templateObject6$2 = _taggedTemplateLiteralLoose(["\n  border-bottom: 1px solid #eeedf1;\n  margin: 2px 0px;\n  width: 100%;\n"])));

var BoatCardDetails = function BoatCardDetails(props) {
  var description = props.description,
    startDate = props.startDate,
    endDate = props.endDate,
    price = props.price,
    ticketNumber = props.ticketNumber,
    phoneNumber = props.phoneNumber,
    termsAndConditions = props.termsAndConditions;
  return React__default.createElement(Container$1, null, React__default.createElement(Section, null, React__default.createElement(Title, null, "description"), React__default.createElement(Value, null, description)), React__default.createElement(Divider, null), React__default.createElement(Section, null, React__default.createElement(Title, null, "departure"), React__default.createElement(Value, null, startDate)), React__default.createElement(Divider, null), React__default.createElement(Section, null, React__default.createElement(Title, null, "arriving at"), React__default.createElement(Value, null, endDate)), React__default.createElement(Divider, null), React__default.createElement(Section, null, React__default.createElement(Title, null, "ticket number"), React__default.createElement(Value, null, ticketNumber)), React__default.createElement(Divider, null), React__default.createElement(Section, null, React__default.createElement(Title, null, "price"), React__default.createElement(Value, null, price)), React__default.createElement(Divider, null), React__default.createElement(Section, null, React__default.createElement(Title, null, "terms & conditions"), React__default.createElement(Value, null, termsAndConditions)), React__default.createElement(Divider, null), React__default.createElement(Section, null, React__default.createElement(Title, null, "customer service number"), React__default.createElement(Value, {
    color: '#3775d1'
  }, phoneNumber)));
};

var CinemaCardDetails = function CinemaCardDetails(props) {
  var movieName = props.movieName,
    rating = props.rating,
    customTitle = props.customTitle,
    address = props.address,
    openTime = props.openTime,
    startDate = props.startDate,
    endDate = props.endDate,
    ticketNumber = props.ticketNumber,
    termsAndConditions = props.termsAndConditions,
    email = props.email,
    phoneNumber = props.phoneNumber;
  return React__default.createElement(Container$1, null, React__default.createElement(Section, null, React__default.createElement(Title, {
    textTransform: 'none'
  }, "Movie Name"), React__default.createElement(Value, null, movieName)), React__default.createElement(Divider, null), React__default.createElement(Section, null, React__default.createElement(Title, {
    textTransform: 'none'
  }, "Rating"), React__default.createElement(Value, null, rating)), React__default.createElement(Divider, null), React__default.createElement(Section, null, React__default.createElement(Title, {
    textTransform: 'none'
  }, customTitle), React__default.createElement(Value, null, address)), React__default.createElement(Divider, null), React__default.createElement(Section, null, React__default.createElement(Title, {
    textTransform: 'none'
  }, "Doors Open"), React__default.createElement(Value, null, openTime)), React__default.createElement(Divider, null), React__default.createElement(Section, null, React__default.createElement(Title, {
    textTransform: 'none'
  }, "Start Date"), React__default.createElement(Value, null, startDate)), React__default.createElement(Divider, null), React__default.createElement(Section, null, React__default.createElement(Title, {
    textTransform: 'none'
  }, "End Date"), React__default.createElement(Value, null, endDate)), React__default.createElement(Divider, null), React__default.createElement(Section, null, React__default.createElement(Title, {
    textTransform: 'none'
  }, "Ticket Number"), React__default.createElement(Value, null, ticketNumber)), React__default.createElement(Divider, null), React__default.createElement(Section, null, React__default.createElement(Title, {
    textTransform: 'none'
  }, "Terms & Conditions"), React__default.createElement(Value, null, termsAndConditions)), React__default.createElement(Divider, null), React__default.createElement(Section, null, React__default.createElement(Title, {
    textTransform: 'none'
  }, "Info Line"), React__default.createElement(Value, {
    color: '#3775d1'
  }, phoneNumber)), React__default.createElement(Divider, null), React__default.createElement(Section, null, React__default.createElement(Title, {
    textTransform: 'none'
  }, "Email"), React__default.createElement(Value, {
    color: '#3775d1'
  }, email)));
};

var CommonCardDetails = function CommonCardDetails(props) {
  var description = props.description,
    termsAndConditions = props.termsAndConditions;
  return React__default.createElement(Container$1, null, React__default.createElement(Section, null, React__default.createElement(Title, null, "details"), React__default.createElement(Value, null, description)), React__default.createElement(Divider, null), React__default.createElement(Section, null, React__default.createElement(Title, null, "terms & conditions"), React__default.createElement(Value, null, termsAndConditions)));
};

var CouponCardDetails = function CouponCardDetails(props) {
  var description = props.description,
    promoCode = props.promoCode,
    couponId = props.couponId;
  return React__default.createElement(Container$1, null, React__default.createElement(Section, null, React__default.createElement(Title, {
    textTransform: 'none'
  }, "Description"), React__default.createElement(Value, null, description)), React__default.createElement(Divider, null), React__default.createElement(Section, null, React__default.createElement(Title, {
    textTransform: 'none'
  }, "Promo Code"), React__default.createElement(Value, null, promoCode)), React__default.createElement(Divider, null), React__default.createElement(Section, null, React__default.createElement(Title, {
    textTransform: 'none'
  }, "Coupon ID"), React__default.createElement(Value, null, couponId)));
};

var EventCardDetails = function EventCardDetails(props) {
  var eventName = props.eventName,
    address = props.address,
    openTime = props.openTime,
    startDate = props.startDate,
    endDate = props.endDate,
    ticketNumber = props.ticketNumber,
    price = props.price,
    termsAndConditions = props.termsAndConditions,
    googleMapsLink = props.googleMapsLink,
    phoneNumber = props.phoneNumber,
    email = props.email;
  return React__default.createElement(Container$1, null, React__default.createElement(Section, null, React__default.createElement(Title, null, eventName), React__default.createElement(Value, null, address)), React__default.createElement(Divider, null), React__default.createElement(Section, null, React__default.createElement(Title, null, "doors open"), React__default.createElement(Value, null, openTime)), React__default.createElement(Divider, null), React__default.createElement(Section, null, React__default.createElement(Title, null, "start date"), React__default.createElement(Value, null, startDate)), React__default.createElement(Divider, null), React__default.createElement(Section, null, React__default.createElement(Title, null, "end date"), React__default.createElement(Value, null, endDate)), React__default.createElement(Divider, null), React__default.createElement(Section, null, React__default.createElement(Title, null, "ticket number"), React__default.createElement(Value, null, ticketNumber)), React__default.createElement(Divider, null), React__default.createElement(Section, null, React__default.createElement(Title, null, "value"), React__default.createElement(Value, null, price)), React__default.createElement(Divider, null), React__default.createElement(Section, null, React__default.createElement(Title, null, "terms & conditions"), React__default.createElement(Value, null, termsAndConditions)), React__default.createElement(Section, null, React__default.createElement(Title, null, "google map"), React__default.createElement(Value, {
    color: '#3775d1'
  }, googleMapsLink)), React__default.createElement(Section, null, React__default.createElement(Title, null, "info line"), React__default.createElement(Value, {
    color: '#3775d1'
  }, phoneNumber)), React__default.createElement(Section, null, React__default.createElement(Title, null, "email"), React__default.createElement(Value, {
    color: '#3775d1'
  }, email)));
};

var PlaneCardDetails = function PlaneCardDetails(props) {
  var description = props.description,
    confirmationCode = props.confirmationCode;
  return React__default.createElement(Container$1, null, React__default.createElement(Section, null, React__default.createElement(Title, null, "flight details"), React__default.createElement(Value, null, description)), React__default.createElement(Divider, null), React__default.createElement(Section, null, React__default.createElement(Title, null, "confirmation code"), React__default.createElement(Value, null, confirmationCode)));
};

var RoomCardDetails = function RoomCardDetails(props) {
  var holderName = props.holderName,
    kind = props.kind,
    address = props.address,
    startDate = props.startDate,
    endDate = props.endDate,
    ticketNumber = props.ticketNumber,
    termsAndConditions = props.termsAndConditions,
    email = props.email,
    phoneNumber = props.phoneNumber;
  return React__default.createElement(Container$1, null, React__default.createElement(Section, null, React__default.createElement(Title, null, "check in"), React__default.createElement(Value, null, startDate)), React__default.createElement(Divider, null), React__default.createElement(Section, null, React__default.createElement(Title, null, "check out"), React__default.createElement(Value, null, endDate)), React__default.createElement(Divider, null), React__default.createElement(Section, null, React__default.createElement(Title, null, "type"), React__default.createElement(Value, null, kind)), React__default.createElement(Divider, null), React__default.createElement(Section, null, React__default.createElement(Title, null, "address"), React__default.createElement(Value, null, address)), React__default.createElement(Divider, null), React__default.createElement(Section, null, React__default.createElement(Title, null, "ticket holder"), React__default.createElement(Value, null, holderName)), React__default.createElement(Divider, null), React__default.createElement(Section, null, React__default.createElement(Title, null, "ticket number"), React__default.createElement(Value, null, ticketNumber)), React__default.createElement(Divider, null), React__default.createElement(Section, null, React__default.createElement(Title, null, "Terms & Conditions"), React__default.createElement(Value, null, termsAndConditions)), React__default.createElement(Divider, null), React__default.createElement(Section, null, React__default.createElement(Title, null, "Info Line"), React__default.createElement(Value, {
    color: '#3775d1'
  }, phoneNumber)), React__default.createElement(Divider, null), React__default.createElement(Section, null, React__default.createElement(Title, {
    textTransform: 'none'
  }, "Email"), React__default.createElement(Value, {
    color: '#3775d1'
  }, email)));
};

var StoreCardDetails = function StoreCardDetails(props) {
  var customTitle = props.customTitle,
    customTitlePhone = props.customTitlePhone,
    description = props.description,
    holderLastName = props.holderLastName,
    holderName = props.holderName,
    kind = props.kind,
    address = props.address,
    schedule = props.schedule,
    accessCodeId = props.accessCodeId,
    offerId = props.offerId,
    couponId = props.couponId,
    phoneNumber = props.phoneNumber,
    phoneNumberSecondary = props.phoneNumberSecondary,
    website = props.website;
  return React__default.createElement(Container$1, null, React__default.createElement(Section, null, React__default.createElement(Title, null, customTitle), React__default.createElement(Value, null, description)), React__default.createElement(Divider, null), React__default.createElement(Section, null, React__default.createElement(Title, null, holderName), React__default.createElement(Value, null, holderLastName)), React__default.createElement(Divider, null), React__default.createElement(Section, null, React__default.createElement(Title, null, "type"), React__default.createElement(Value, null, kind)), React__default.createElement(Divider, null), React__default.createElement(Section, null, React__default.createElement(Title, null, "address"), React__default.createElement(Value, null, address)), React__default.createElement(Divider, null), React__default.createElement(Section, null, React__default.createElement(Title, null, "schedule"), React__default.createElement(Value, null, schedule)), React__default.createElement(Divider, null), React__default.createElement(Section, null, React__default.createElement(Title, null, "barcode"), React__default.createElement(Value, null, accessCodeId)), React__default.createElement(Divider, null), React__default.createElement(Section, null, React__default.createElement(Title, null, "offers"), React__default.createElement(Value, null, offerId)), React__default.createElement(Divider, null), React__default.createElement(Section, null, React__default.createElement(Title, null, "coupons"), React__default.createElement(Value, null, couponId)), React__default.createElement(Divider, null), React__default.createElement(Section, null, React__default.createElement(Title, null, customTitlePhone), React__default.createElement(Row, null, React__default.createElement(Value, null, "Commercial phone :"), React__default.createElement(Value, {
    color: '#3775d1'
  }, phoneNumber)), React__default.createElement(Row, null, React__default.createElement(Value, null, "After sale service :"), React__default.createElement(Value, {
    color: '#3775d1'
  }, phoneNumberSecondary))), React__default.createElement(Divider, null), React__default.createElement(Section, null, React__default.createElement(Title, null, "website"), React__default.createElement(Value, {
    color: '#3775d1'
  }, website)));
};

var _templateObject$5, _templateObject2$5, _templateObject3$4, _templateObject4$4, _templateObject5$3, _templateObject6$3, _templateObject7$2, _templateObject8, _templateObject9, _templateObject10, _templateObject11, _templateObject12, _templateObject13, _templateObject14, _templateObject15, _templateObject16, _templateObject17, _templateObject18, _templateObject19;
var CardContainer = styled("div")(_templateObject$5 || (_templateObject$5 = _taggedTemplateLiteralLoose(["\n  margin: 0px auto;\n  max-width: 340px;\n  min-width: 350px;\n"])));
var Card = styled("div")(_templateObject2$5 || (_templateObject2$5 = _taggedTemplateLiteralLoose(["\n  perspective: 1000px;\n"])));
var Content = styled("div")(_templateObject3$4 || (_templateObject3$4 = _taggedTemplateLiteralLoose(["\n  position: relative;\n  transform: ", ";\n  transition: transform ease 0.6s;\n  transform-style: preserve-3d;\n"])), function (props) {
  return props.flipped ? "rotateY(180deg)" : "rotateY(0)";
});
var CardFrontRegular = styled("div")(_templateObject4$4 || (_templateObject4$4 = _taggedTemplateLiteralLoose(["\n  display: flex;\n  flex-direction: column;\n  justify-content: center;\n  position: absolute;\n  overflow: hidden;\n  -webkit-backface-visibility: hidden;\n  backface-visibility: hidden;\n  box-sizing: border-box;\n  height: fit-content;\n  min-width: 330px;\n  max-width: 340px;\n  background: ", ";\n  border: 1px solid #f2ecec;\n  border-radius: 10px;\n"])), function (props) {
  return props.backgroundColor || "#ffffff";
});
var CardBackRegular = styled("div")(_templateObject5$3 || (_templateObject5$3 = _taggedTemplateLiteralLoose(["\n  display: flex;\n  flex-direction: column;\n  justify-content: center;\n  position: ", ";\n  overflow: hidden;\n  -webkit-backface-visibility: hidden;\n  backface-visibility: hidden;\n  box-sizing: border-box;\n  height: fit-content;\n  width: fit-content;\n  padding: 9px;\n  background: #ffffff;\n  border: 1px solid #f2ecec;\n  border-radius: 10px;\n  transform: rotateY(180deg);\n  transition: transform ease 0.6s;\n"])), function (props) {
  return props.flipped ? "unset" : "absolute";
});
var CardFrontCoupon = styled("div")(_templateObject6$3 || (_templateObject6$3 = _taggedTemplateLiteralLoose(["\n  display: flex;\n  flex-direction: column;\n  justify-content: center;\n  position: absolute;\n  overflow: hidden;\n  -webkit-backface-visibility: hidden;\n  backface-visibility: hidden;\n  box-sizing: border-box;\n  height: fit-content;\n  width: fit-content;\n  min-height: 450px;\n  min-width: 330px;\n  max-width: 340px;\n  margin: 0 auto;\n  background: ", ";\n  border: 1px solid #f2ecec;\n  --mask: linear-gradient(#f2ecec 0 0) center/100% calc(100% - 8px) no-repeat,\n    radial-gradient(circle farthest-side, #0000 100%, #f2ecec) 0.2% -9px/12px 13px\n      repeat-x,\n    radial-gradient(circle farthest-side, #0000 100%, #f2ecec) 0.2%\n      calc(100% + 9px) / 12px 13px repeat-x;\n  mask: var(--mask);\n"])), function (props) {
  return props.backgroundColor || "#ffffff";
});
var CardBackCoupon = styled("div")(_templateObject7$2 || (_templateObject7$2 = _taggedTemplateLiteralLoose(["\n  display: flex;\n  flex-direction: column;\n  position: ", ";\n  overflow: hidden;\n  -webkit-backface-visibility: hidden;\n  backface-visibility: hidden;\n  box-sizing: border-box;\n  height: fit-content;\n  width: fit-content;\n  min-height: 500px;\n  min-width: 330px;\n  max-width: 335px;\n  margin: 0 auto;\n  padding: 9px;\n  border: 1px solid #f2ecec;\n  background: ", ";\n  transform: rotateY(180deg);\n  transition: transform ease 0.6s;\n  --mask: linear-gradient(#000 0 0) center/100% calc(100% - 8px) no-repeat,\n    radial-gradient(circle farthest-side, #0000 100%, #000) 0.2% -9px/12px 13px repeat-x,\n    radial-gradient(circle farthest-side, #0000 100%, #000) 0.2%\n      calc(100% + 9px) / 12px 13px repeat-x;\n  mask: var(--mask);\n"])), function (props) {
  return props.flipped ? "unset" : "absolute";
}, function (props) {
  return props.backgroundColor || "#ffffff";
});
var CardFrontSemiCirlce = styled("div")(_templateObject8 || (_templateObject8 = _taggedTemplateLiteralLoose(["\n  display: flex;\n  flex-direction: column;\n  justify-content: center;\n  position: absolute;\n  overflow: hidden;\n  -webkit-backface-visibility: hidden;\n  backface-visibility: hidden;\n  box-sizing: border-box;\n  height: fit-content;\n  width: fit-content;\n  min-height: 350px;\n  min-width: 330px;\n  max-width: 340px;\n  margin: 0 auto;\n  background: ", ";\n  border: 1px solid #f2ecec;\n"])), function (props) {
  return props.backgroundColor || "#ffffff";
});
var CardBackSemiCircle = styled("div")(_templateObject9 || (_templateObject9 = _taggedTemplateLiteralLoose(["\n  display: flex;\n  flex-direction: column;\n  justify-content: center;\n  position: ", ";\n  overflow: hidden;\n  -webkit-backface-visibility: hidden;\n  backface-visibility: hidden;\n  box-sizing: border-box;\n  height: fit-content;\n  width: fit-content;\n  min-height: 350px;\n  min-width: 350px;\n  margin: 0 auto;\n  padding: 9px;\n  border: 1px solid #f2ecec;\n  background: #ffffff;\n  transform: rotateY(180deg);\n  transition: transform ease 0.6s;\n"])), function (props) {
  return props.flipped ? "unset" : "absolute";
});
var CardFrontDoubleCirlce = styled("div")(_templateObject10 || (_templateObject10 = _taggedTemplateLiteralLoose(["\n  display: flex;\n  flex-direction: column;\n  justify-content: center;\n  position: absolute;\n  overflow: hidden;\n  -webkit-backface-visibility: hidden;\n  backface-visibility: hidden;\n  box-sizing: border-box;\n  height: fit-content;\n  width: fit-content;\n  min-height: 350px;\n  min-width: 330px;\n  max-width: 340px;\n  margin: 0 auto;\n  background: ", ";\n  border: 1px solid #f2ecec;\n  border-radius: 10px;\n"])), function (props) {
  return props.backgroundColor || "#ffffff";
});
var CardBackDoubleCircle = styled("div")(_templateObject11 || (_templateObject11 = _taggedTemplateLiteralLoose(["\n  display: flex;\n  flex-direction: column;\n  position: ", ";\n  overflow: hidden;\n  -webkit-backface-visibility: hidden;\n  backface-visibility: hidden;\n  box-sizing: border-box;\n  width: fit-content;\n  min-width: 350px;\n  min-height: 525px;\n  margin: 0 auto;\n  padding: 12px;\n  border: 1px solid #f2ecec;\n  border-radius: 10px;\n  background: #ffffff;\n  transform: rotateY(180deg);\n  transition: transform ease 0.6s;\n"])), function (props) {
  return props.flipped ? "unset" : "absolute";
});
var Button = styled("button")(_templateObject12 || (_templateObject12 = _taggedTemplateLiteralLoose(["\n  display: flex;\n  flex-direction: column;\n  align-items: center;\n  justify-content: center;\n  width: 30px;\n  height: 30px;\n  border: none;\n  border-radius: 15px;\n  position: absolute;\n  z-index: 999;\n  right: 5px;\n  bottom: 0px;\n  background: transparent;\n  :hover {\n    background: #0000000a;\n  }\n"])));
var ContainerSemiCircle = styled("div")(_templateObject13 || (_templateObject13 = _taggedTemplateLiteralLoose(["\n  position: absolute;\n  z-index: 99;\n  top: 0px;\n  left: calc(50% - 35px);\n  width: 70px;\n  height: 16px;\n  border-top-left-radius: 35px;\n  border-top-right-radius: 35px;\n  transform: rotate(180deg);\n  overflow: hidden;\n"])));
var SemiCircle = styled("div")(_templateObject14 || (_templateObject14 = _taggedTemplateLiteralLoose(["\n  width: 60px;\n  height: 60px;\n  margin: auto;\n  border-radius: 30px;\n  border: 1px solid #f2ecec;\n  background-color: ", ";\n"])), function (props) {
  return props.backgroundColor;
});
var ContainerLeftSemiCircle = styled("div")(_templateObject15 || (_templateObject15 = _taggedTemplateLiteralLoose(["\n  position: absolute;\n  z-index: 1;\n  top: 120px;\n  left: -31px;\n  width: 70px;\n  height: 8px;\n  border-top-left-radius: 35px;\n  border-top-right-radius: 35px;\n  transform: rotate(90deg);\n  overflow: hidden;\n"])));
var SemiLeftCircle = styled("div")(_templateObject16 || (_templateObject16 = _taggedTemplateLiteralLoose(["\n  width: 20px;\n  height: 25px;\n  margin: auto;\n  border-radius: 30px;\n  border: 1px solid #f2ecec;\n  background-color: ", ";\n"])), function (props) {
  return props.backgroundColor;
});
var ContainerRightBackSemiCircle = styled("div")(_templateObject17 || (_templateObject17 = _taggedTemplateLiteralLoose(["\n  position: absolute;\n  z-index: 1;\n  top: 120px;\n  right: ", ";\n  width: 70px;\n  height: 8px;\n  border-top-left-radius: 35px;\n  border-top-right-radius: 35px;\n  transform: rotate(270deg);\n  overflow: hidden;\n  display: ", ";\n"])), function (props) {
  return props.type === "plane" ? "-40.5px" : "-31px";
}, function (props) {
  return props.flipped ? "block" : "none";
});
var ContainerRightFrontSemiCircle = styled("div")(_templateObject18 || (_templateObject18 = _taggedTemplateLiteralLoose(["\n  position: absolute;\n  z-index: 1;\n  top: 120px;\n  right: ", ";\n  width: 70px;\n  height: 8px;\n  border-top-left-radius: 35px;\n  border-top-right-radius: 35px;\n  transform: rotate(270deg);\n  overflow: hidden;\n  display: ", ";\n"])), function (props) {
  return props.type === "plane" ? "-40.5px" : "-50.5px";
}, function (props) {
  return props.flipped ? "none" : "block";
});
var SemiRightCircle = styled("div")(_templateObject19 || (_templateObject19 = _taggedTemplateLiteralLoose(["\n  width: 20px;\n  height: 25px;\n  margin: auto;\n  border-radius: 30px;\n  border: 1px solid #f2ecec;\n  background-color: ", ";\n"])), function (props) {
  return props.backgroundColor;
});

var CardIOS = function CardIOS(props) {
  var backgroundColor = props.backgroundColor,
    type = props.type;
  var _useState = React.useState(false),
    flipped = _useState[0],
    setFlipped = _useState[1];
  var onClickFlip = function onClickFlip() {
    setFlipped(!flipped);
  };
  var renderButton = function renderButton(position) {
    if (position === 'front' && !flipped || position === 'back' && flipped) {
      return React__default.createElement(Button, {
        onClick: onClickFlip
      }, React__default.createElement(Svgs.MoreIcon, {
        color: '#0000008a'
      }));
    }
    return null;
  };
  var renderSemiCircle = function renderSemiCircle() {
    return React__default.createElement(ContainerSemiCircle, null, React__default.createElement(SemiCircle, {
      backgroundColor: '#FFFFFF'
    }));
  };
  var renderSemiLeftCircle = function renderSemiLeftCircle() {
    return React__default.createElement(ContainerLeftSemiCircle, null, React__default.createElement(SemiLeftCircle, {
      backgroundColor: '#FFFFFF'
    }));
  };
  var renderSemiRightFrontCircle = function renderSemiRightFrontCircle() {
    return React__default.createElement(ContainerRightFrontSemiCircle, {
      flipped: flipped,
      type: type
    }, React__default.createElement(SemiRightCircle, {
      backgroundColor: '#FFFFFF'
    }));
  };
  var renderSemiRightBackCircle = function renderSemiRightBackCircle() {
    return React__default.createElement(ContainerRightBackSemiCircle, {
      flipped: flipped
    }, React__default.createElement(SemiRightCircle, {
      backgroundColor: '#FFFFFF'
    }));
  };
  var renderHeader = function renderHeader() {
    if (type === 'store' || type === 'coupon' || type === 'plane') {
      return React__default.createElement(Header, {
        headerLogoLink: props.headerLogoLink,
        headerTitle: props.headerTitle,
        title: props.title,
        mainTitle: props.mainTitle
      });
    }
    if (type === 'event' || type === 'cinema' || type === 'room') {
      return React__default.createElement(Header, {
        headerLogoLink: props.headerLogoLink,
        headerTitle: props.headerTitle
      });
    }
    if (type === 'boat' || type === 'bus' || type === 'subway' || type === 'train') {
      return React__default.createElement(Header, {
        headerLogoLink: props.headerLogoLink,
        title: props.title,
        mainTitle: props.mainTitle
      });
    } else return null;
  };
  var renderBanner = function renderBanner() {
    if (type === 'store' || type === 'event' || type === 'coupon' || type === 'room' || type === 'cinema') {
      return React__default.createElement(Banner, {
        bannerImage: props.bannerImage
      });
    }
    return null;
  };
  var renderInformation = function renderInformation() {
    if (type === 'store' || type === 'coupon') {
      return React__default.createElement(Information, {
        type: props.type,
        bottomLeftTitle: props.bottomLeftTitle,
        bottomLeftValue: props.bottomLeftValue,
        bottomRightTitle: props.bottomRightTitle,
        bottomRightValue: props.bottomRightValue
      });
    }
    if (type === 'event') {
      return React__default.createElement(Information, {
        type: props.type,
        topLeftTitle: props.topLeftTitle,
        topLeftValue: props.topLeftValue,
        topCenterLeftTitle: props.topCenterLeftTitle,
        topCenterLeftValue: props.topCenterLeftValue,
        topRightTitle: props.topRightTitle,
        topRightValue: props.topRightValue,
        bottomLeftTitle: props.bottomLeftTitle,
        bottomLeftValue: props.bottomLeftValue,
        bottomRightTitle: props.bottomRightTitle,
        bottomRightValue: props.bottomRightValue
      });
    }
    if (type === 'plane') {
      return React__default.createElement(Information, {
        type: props.type,
        LeftTitle: props.LeftTitle,
        LeftValue: props.LeftValue,
        RightTitle: props.RightTitle,
        RightValue: props.RightValue,
        topLeftTitle: props.topLeftTitle,
        topLeftValue: props.topLeftValue,
        topCenterLeftTitle: props.topCenterLeftTitle,
        topCenterLeftValue: props.topCenterLeftValue,
        topCenterRightTitle: props.topCenterRightTitle,
        topCenterRightValue: props.topCenterRightValue,
        topRightTitle: props.topRightTitle,
        topRightValue: props.topRightValue,
        bottomLeftTitle: props.bottomLeftTitle,
        bottomLeftValue: props.bottomLeftValue,
        bottomRightTitle: props.bottomRightTitle,
        bottomRightValue: props.bottomRightValue
      });
    }
    if (type === 'boat' || type === 'bus' || type === 'subway' || type === 'train') {
      return React__default.createElement(Information, {
        type: props.type,
        LeftTitle: props.LeftTitle,
        LeftValue: props.LeftValue,
        RightTitle: props.RightTitle,
        RightValue: props.RightValue,
        topLeftTitle: props.topLeftTitle,
        topLeftValue: props.topLeftValue,
        topCenterLeftTitle: props.topCenterLeftTitle,
        topCenterLeftValue: props.topCenterLeftValue,
        topRightTitle: props.topRightTitle,
        topRightValue: props.topRightValue,
        bottomLeftTitle: props.bottomLeftTitle,
        bottomLeftValue: props.bottomLeftValue
      });
    }
    if (type === 'room') {
      return React__default.createElement(Information, {
        type: props.type,
        topLeftTitle: props.topLeftTitle,
        topLeftValue: props.topLeftValue,
        bottomLeftTitle: props.bottomLeftTitle,
        bottomLeftValue: props.bottomLeftValue,
        bottomCenterTitle: props.bottomCenterTitle,
        bottomCenterValue: props.bottomCenterValue,
        bottomRightTitle: props.bottomRightTitle,
        bottomRightValue: props.bottomRightValue
      });
    }
    if (type === 'cinema') {
      return React__default.createElement(Information, {
        type: props.type,
        topLeftTitle: props.topLeftTitle,
        topLeftValue: props.topLeftValue,
        topCenterLeftTitle: props.topCenterLeftTitle,
        topCenterLeftValue: props.topCenterLeftValue,
        topCenterRightTitle: props.topCenterRightTitle,
        topCenterRightValue: props.topCenterRightValue,
        topRightTitle: props.topRightTitle,
        topRightValue: props.topRightValue,
        bottomLeftTitle: props.bottomLeftTitle,
        bottomLeftValue: props.bottomLeftValue,
        bottomRightTitle: props.bottomRightTitle,
        bottomRightValue: props.bottomRightValue
      });
    } else return null;
  };
  var renderAccess = function renderAccess() {
    if (type) {
      return React__default.createElement(Access, {
        accessImage: props.accessImage,
        accessType: props.accessType,
        accessCodeId: props.accessCodeId
      });
    } else return null;
  };
  var renderDetails = function renderDetails() {
    if (type === 'store') {
      return React__default.createElement(StoreCardDetails, {
        customTitle: props.customTitle,
        customTitlePhone: props.customTitlePhone,
        description: props.description,
        holderLastName: props.holderLastName,
        holderName: props.holderName,
        kind: props.kind,
        address: props.address,
        schedule: props.schedule,
        accessCodeId: props.accessCodeId,
        offerId: props.offerId,
        couponId: props.couponId,
        phoneNumber: props.phoneNumber,
        phoneNumberSecondary: props.phoneNumberSecondary,
        website: props.website
      });
    }
    if (type === 'event') {
      return React__default.createElement(EventCardDetails, {
        eventName: props.eventName,
        address: props.address,
        openTime: props.openTime,
        startDate: props.startDate,
        endDate: props.endDate,
        ticketNumber: props.ticketNumber,
        price: props.price,
        termsAndConditions: props.termsAndConditions,
        googleMapsLink: props.googleMapsLink,
        phoneNumber: props.phoneNumber,
        email: props.email
      });
    }
    if (type === 'coupon') {
      return React__default.createElement(CouponCardDetails, {
        description: props.description,
        promoCode: props.promoCode,
        couponId: props.couponId
      });
    }
    if (type === 'plane') {
      return React__default.createElement(PlaneCardDetails, {
        description: props.description,
        confirmationCode: props.confirmationCode
      });
    }
    if (type === 'boat') {
      return React__default.createElement(BoatCardDetails, {
        description: props.description,
        startDate: props.startDate,
        endDate: props.endDate,
        ticketNumber: props.ticketNumber,
        price: props.price,
        termsAndConditions: props.termsAndConditions,
        phoneNumber: props.phoneNumber
      });
    }
    if (type === 'cinema') {
      return React__default.createElement(CinemaCardDetails, {
        movieName: props.movieName,
        rating: props.rating,
        customTitle: props.customTitle,
        openTime: props.openTime,
        address: props.address,
        startDate: props.startDate,
        endDate: props.endDate,
        ticketNumber: props.ticketNumber,
        termsAndConditions: props.termsAndConditions,
        phoneNumber: props.phoneNumber,
        email: props.email
      });
    }
    if (type === 'room') {
      return React__default.createElement(RoomCardDetails, {
        startDate: props.startDate,
        endDate: props.endDate,
        kind: props.kind,
        address: props.address,
        holderName: props.holderName,
        ticketNumber: props.ticketNumber,
        termsAndConditions: props.termsAndConditions,
        phoneNumber: props.phoneNumber,
        email: props.email
      });
    }
    if (type === 'bus' || type === 'subway' || type === 'train') {
      return React__default.createElement(CommonCardDetails, {
        description: props.description,
        termsAndConditions: props.termsAndConditions
      });
    } else return null;
  };
  var renderContent = function renderContent() {
    if (type === 'store') {
      return React__default.createElement(Content, {
        flipped: flipped
      }, React__default.createElement(CardFrontRegular, {
        backgroundColor: backgroundColor
      }, renderHeader(), renderBanner(), renderInformation(), renderAccess(), renderButton('front')), React__default.createElement(CardBackRegular, {
        backgroundColor: backgroundColor,
        flipped: flipped
      }, renderDetails(), renderButton('back')));
    }
    if (type === 'coupon') {
      return React__default.createElement(Content, {
        flipped: flipped
      }, React__default.createElement(CardFrontCoupon, {
        backgroundColor: backgroundColor
      }, renderHeader(), renderBanner(), renderInformation(), renderAccess(), renderButton('front')), React__default.createElement(CardBackCoupon, {
        backgroundColor: backgroundColor,
        flipped: flipped
      }, renderDetails(), renderButton('back')));
    }
    if (type === 'event' || type === 'cinema' || type === 'room') {
      return React__default.createElement(Content, {
        flipped: flipped
      }, renderSemiCircle(), React__default.createElement(CardFrontSemiCirlce, {
        backgroundColor: backgroundColor
      }, renderHeader(), renderBanner(), renderInformation(), renderAccess(), renderButton('front')), React__default.createElement(CardBackSemiCircle, {
        backgroundColor: backgroundColor,
        flipped: flipped
      }, renderDetails(), renderButton('back')));
    }
    if (type === 'plane' || type === 'boat' || type === 'bus' || type === 'subway' || type === 'train') {
      return React__default.createElement(Content, {
        flipped: flipped
      }, renderSemiLeftCircle(), renderSemiRightFrontCircle(), renderSemiRightBackCircle(), React__default.createElement(CardFrontDoubleCirlce, {
        backgroundColor: backgroundColor
      }, renderHeader(), renderBanner(), renderInformation(), renderAccess(), renderButton('front')), React__default.createElement(CardBackDoubleCircle, {
        backgroundColor: backgroundColor,
        flipped: flipped
      }, renderDetails(), renderButton('back')));
    } else return null;
  };
  return React__default.createElement(CardContainer, null, React__default.createElement(Card, null, renderContent()));
};

var _templateObject$6, _templateObject2$6;
var FooterContainer = styled('div')(_templateObject$6 || (_templateObject$6 = _taggedTemplateLiteralLoose(["\n  height: 125px;\n  width: 100%;\n  overflow: hidden;\n  box-sizing: border-box;\n"])));
var Image$1 = styled('img')(_templateObject2$6 || (_templateObject2$6 = _taggedTemplateLiteralLoose(["\n  object-fit: cover;\n  height: inherit;\n  width: 100%;\n"])));

var Footer = function Footer(props) {
  var bannerImage = props.bannerImage;
  if (bannerImage) {
    return React__default.createElement(FooterContainer, null, React__default.createElement(Image$1, {
      src: bannerImage
    }));
  } else return null;
};

var _templateObject$7, _templateObject2$7, _templateObject3$5;
var HeaderContainer$1 = styled('div')(_templateObject$7 || (_templateObject$7 = _taggedTemplateLiteralLoose(["\n  padding: 14px 14px 12px;\n  width: 100%;\n  display: flex;\n  align-items: center;\n  gap: 15px;\n  box-sizing: border-box;\n  justify-content: ", ";\n  position: relative;\n  box-shadow: 0px -6px 10px 0px #000000;\n"])), function (props) {
  return props.justifyContent ? props.justifyContent : 'flex-start';
});
var Title$1 = styled('h4')(_templateObject2$7 || (_templateObject2$7 = _taggedTemplateLiteralLoose(["\n  font-size: 0.95rem;\n  font-weight: 500;\n  color: ", ";\n  letter-spacing: 2px;\n  margin: 0;\n"])), function (props) {
  return props.textColor;
});
var ImageContainer$1 = styled('img')(_templateObject3$5 || (_templateObject3$5 = _taggedTemplateLiteralLoose(["\n  position: relative;\n  display: flex;\n  height: 40px;\n  align-items: center;\n  justify-content: center;\n  border: 1px solid #f2ecec;\n  border-radius: 50%;\n  img {\n    height: 32px;\n    width: 32px;\n    object-fit: contain;\n  }\n"])));

var Header$1 = function Header(props) {
  var titleColor = props.titleColor,
    headerLogoLink = props.headerLogoLink,
    title = props.title;
  return React__default.createElement(HeaderContainer$1, null, React__default.createElement(ImageContainer$1, {
    src: headerLogoLink
  }), React__default.createElement(Title$1, {
    textColor: titleColor
  }, title));
};

var _templateObject$8, _templateObject2$8, _templateObject3$6, _templateObject4$5;
var Container$2 = styled("div")(_templateObject$8 || (_templateObject$8 = _taggedTemplateLiteralLoose(["\n  padding: 0px 0px 12px;\n  width: 100%;\n  display: flex;\n  flex-direction: column;\n  align-items: flex-start;\n  gap: 3px;\n  box-sizing: border-box;\n"])));
var Row$1 = styled("div")(_templateObject2$8 || (_templateObject2$8 = _taggedTemplateLiteralLoose(["\n  display: flex;\n  align-items: center;\n  gap: ", ";\n"])), function (_ref) {
  var gap = _ref.gap;
  return gap || "4px";
});
var SecondaryTitle = styled("h4")(_templateObject3$6 || (_templateObject3$6 = _taggedTemplateLiteralLoose(["\n  font-size: 14px;\n  font-weight: 700;\n  color: rgb(0, 0, 0);\n  margin: 0;\n"])));
var MainTitle = styled("h4")(_templateObject4$5 || (_templateObject4$5 = _taggedTemplateLiteralLoose(["\n  font-size: 20px;\n  font-weight: 700;\n  color: rgb(0, 0, 0);\n  margin: 0;\n"])));

var CardTitle = function CardTitle(props) {
  var composedMainTitle = props.composedMainTitle,
    composedSecondaryTitle = props.composedSecondaryTitle,
    secondaryTitle = props.secondaryTitle,
    mainTitle = props.mainTitle,
    type = props.type;
  return React__default.createElement(Container$2, null, secondaryTitle && React__default.createElement(SecondaryTitle, null, secondaryTitle), React__default.createElement(MainTitle, null, mainTitle), (composedSecondaryTitle === null || composedSecondaryTitle === void 0 ? void 0 : composedSecondaryTitle.from) && (composedSecondaryTitle === null || composedSecondaryTitle === void 0 ? void 0 : composedSecondaryTitle.to) && React__default.createElement(Row$1, {
    gap: "4px",
    alignItems: "flex-start"
  }, React__default.createElement(SecondaryTitle, null, composedSecondaryTitle === null || composedSecondaryTitle === void 0 ? void 0 : composedSecondaryTitle.from), React__default.createElement("span", null, "-"), React__default.createElement(SecondaryTitle, null, composedSecondaryTitle === null || composedSecondaryTitle === void 0 ? void 0 : composedSecondaryTitle.to)), (composedMainTitle === null || composedMainTitle === void 0 ? void 0 : composedMainTitle.from) && (composedMainTitle === null || composedMainTitle === void 0 ? void 0 : composedMainTitle.to) && React__default.createElement(Row$1, {
    gap: "12px"
  }, React__default.createElement(MainTitle, null, composedMainTitle === null || composedMainTitle === void 0 ? void 0 : composedMainTitle.from), type === "plane" && React__default.createElement(Svgs.PlaneIcon, null), (type === "boat" || type === "bus" || type === "subway" || type === "train") && React__default.createElement(Svgs.ArrowIcon, null), React__default.createElement(MainTitle, null, composedMainTitle === null || composedMainTitle === void 0 ? void 0 : composedMainTitle.to)));
};

var _templateObject$9, _templateObject2$9, _templateObject3$7, _templateObject4$6, _templateObject5$4, _templateObject6$4;
var InfosContainer = styled("div")(_templateObject$9 || (_templateObject$9 = _taggedTemplateLiteralLoose([""])));
var Wrapper = styled("div")(_templateObject2$9 || (_templateObject2$9 = _taggedTemplateLiteralLoose(["\n  display: flex;\n  justify-content: space-between;\n  padding: 0px;\n"])));
var InfoContainer = styled("div")(_templateObject3$7 || (_templateObject3$7 = _taggedTemplateLiteralLoose(["\n  display: flex;\n  flex-direction: column;\n  gap: 2px;\n  text-align: ", ";\n"])), function (_ref) {
  var textAlign = _ref.textAlign;
  return textAlign || "left";
});
var InfoTitle = styled("p")(_templateObject4$6 || (_templateObject4$6 = _taggedTemplateLiteralLoose(["\n  font-size: 12px;\n  font-weight: 700;\n  color: rgb(0, 0, 0);\n  letter-spacing: 1px;\n  text-transform: capitalize;\n  margin: 0px;\n"])));
var InfoValue = styled("p")(_templateObject5$4 || (_templateObject5$4 = _taggedTemplateLiteralLoose(["\n  font-size: 16px;\n  color: rgb(0, 0, 0);\n  margin: 0px;\n  font-weight: 600;\n"])));
var Seperator = styled("div")(_templateObject6$4 || (_templateObject6$4 = _taggedTemplateLiteralLoose(["\n  height: 1px;\n  background-color: rgb(219, 219, 219);\n  border-radius: 1px;\n  width: 100%;\n  margin: 14px 0px;\n"])));

var Infos = function Infos(props) {
  var topLeftTitle = props.topLeftTitle,
    topLeftValue = props.topLeftValue,
    topCenterTitle = props.topCenterTitle,
    topCenterValue = props.topCenterValue,
    topRightTitle = props.topRightTitle,
    topRightValue = props.topRightValue,
    bottomLeftTitle = props.bottomLeftTitle,
    bottomLeftValue = props.bottomLeftValue,
    bottomCenterTitle = props.bottomCenterTitle,
    bottomCenterValue = props.bottomCenterValue,
    bottomRightTitle = props.bottomRightTitle,
    bottomRightValue = props.bottomRightValue;
  return React__default.createElement(InfosContainer, null, React__default.createElement(Wrapper, null, topLeftTitle && topLeftValue && React__default.createElement(InfoContainer, null, React__default.createElement(InfoTitle, null, topLeftTitle), React__default.createElement(InfoValue, null, topLeftValue)), topCenterTitle && topCenterValue && React__default.createElement(InfoContainer, {
    textAlign: 'center'
  }, React__default.createElement(InfoTitle, null, topCenterTitle), React__default.createElement(InfoValue, null, topCenterValue)), topRightTitle && topRightValue && React__default.createElement(InfoContainer, {
    textAlign: 'right'
  }, React__default.createElement(InfoTitle, null, topRightTitle), React__default.createElement(InfoValue, null, topRightValue))), (bottomLeftTitle && bottomLeftValue || bottomCenterTitle && bottomCenterValue || bottomRightTitle && bottomRightValue) && React__default.createElement(Seperator, null), React__default.createElement(Wrapper, null, bottomLeftTitle && bottomLeftValue && React__default.createElement(InfoContainer, null, React__default.createElement(InfoTitle, null, bottomLeftTitle), React__default.createElement(InfoValue, null, bottomLeftValue)), bottomCenterTitle && bottomCenterValue && React__default.createElement(InfoContainer, {
    textAlign: 'center'
  }, React__default.createElement(InfoTitle, null, bottomCenterTitle), React__default.createElement(InfoValue, null, bottomCenterValue)), bottomRightTitle && bottomRightValue && React__default.createElement(InfoContainer, {
    textAlign: 'right'
  }, React__default.createElement(InfoTitle, null, bottomRightTitle), React__default.createElement(InfoValue, null, bottomRightValue))));
};

var _templateObject$a, _templateObject2$a, _templateObject3$8, _templateObject4$7;
var Container$3 = styled("div")(_templateObject$a || (_templateObject$a = _taggedTemplateLiteralLoose(["\n  width: 100%;\n  display: flex;\n  flex-direction: column;\n  justify-content: center;\n  align-items: center;\n  margin: 20px 0 14px;\n  padding: 0 14px;\n  box-sizing: border-box;\n"])));
var ImageContainer$2 = styled("div")(_templateObject2$a || (_templateObject2$a = _taggedTemplateLiteralLoose(["\n  display: flex;\n  height: ", ";\n  width: ", ";\n  -webkit-box-pack: center;\n  justify-content: center;\n  -webkit-box-align: center;\n  align-items: center;\n  background: rgb(255, 255, 255);\n  border-radius: 16px;\n  box-shadow: rgb(0 0 0 / 16%) 0px 1px 1px 0px;\n  box-sizing: border-box;\n"])), function (_ref) {
  var type = _ref.type;
  return type === "qrCode" || type === "aztec" ? "128px" : "100px";
}, function (_ref2) {
  var type = _ref2.type;
  return type === "qrCode" || type === "aztec" ? "128px" : "100%";
});
var Image$2 = styled("img")(_templateObject3$8 || (_templateObject3$8 = _taggedTemplateLiteralLoose(["\n  width: 70%;\n  height: 70px;\n  object-fit: cover;\n"])));
var Text = styled("p")(_templateObject4$7 || (_templateObject4$7 = _taggedTemplateLiteralLoose(["\n  font-family: \"Product Sans\", sans-serif;\n  font-size: 14px;\n  color: rgb(0, 0, 0);\n  text-align: center;\n  letter-spacing: 0.025em;\n  margin-top: 14px;\n  margin-bottom: 0px;\n"])));

var Access$1 = function Access(props) {
  var accessImage = props.accessImage,
    accessType = props.accessType;
  return React__default.createElement(Container$3, null, React__default.createElement(ImageContainer$2, {
    type: accessType
  }, React__default.createElement(Image$2, {
    src: accessImage
  })), props.accessCodeId && React__default.createElement(Text, null, props.accessCodeId));
};

var _templateObject$b, _templateObject2$b, _templateObject3$9, _templateObject4$8, _templateObject5$5;
var Container$4 = styled('div')(_templateObject$b || (_templateObject$b = _taggedTemplateLiteralLoose(["\n  display: flex;\n  flex-direction: column;\n  gap: 15px;\n"])));
var Section$1 = styled('div')(_templateObject2$b || (_templateObject2$b = _taggedTemplateLiteralLoose(["\n  display: flex;\n  flex-direction: ", ";\n  align-items: ", ";\n  gap: ", ";\n"])), function (_ref) {
  var flexDirection = _ref.flexDirection;
  return flexDirection || 'column';
}, function (_ref2) {
  var alignItems = _ref2.alignItems;
  return alignItems || 'flex-start';
}, function (_ref3) {
  var gap = _ref3.gap;
  return gap || '4px';
});
var Row$2 = styled('div')(_templateObject3$9 || (_templateObject3$9 = _taggedTemplateLiteralLoose(["\n  display: flex;\n  gap: 5px;\n"])));
var Title$2 = styled('p')(_templateObject4$8 || (_templateObject4$8 = _taggedTemplateLiteralLoose(["\n  color: rgb(104, 103, 103);\n  font-size: 14px;\n  letter-spacing: 0.075em;\n  line-height: 20px;\n  /* text-transform: uppercase; */\n  margin: 0px;\n"])));
var Value$1 = styled('p')(_templateObject5$5 || (_templateObject5$5 = _taggedTemplateLiteralLoose(["\n  font-size: 14px;\n  letter-spacing: 0.075em;\n  line-height: 20px;\n  color: rgb(0, 0, 0);\n  word-break: break-all;\n  margin: 0px;\n"])));

var StoreCardDetails$1 = function StoreCardDetails(props) {
  var holderName = props.holderName,
    holderId = props.holderId,
    kind = props.kind,
    date = props.date,
    address = props.address,
    customTitle = props.customTitle,
    customValue = props.customValue,
    endDate = props.endDate,
    startDate = props.startDate,
    description = props.description,
    phoneNumber = props.phoneNumber,
    phoneNumberSecondary = props.phoneNumberSecondary,
    website = props.website,
    disableNotifications = props.disableNotifications;
  return React__default.createElement(Container$4, null, React__default.createElement(Section$1, null, React__default.createElement(Title$2, null, "Client name"), React__default.createElement(Value$1, null, holderName)), React__default.createElement(Section$1, null, React__default.createElement(Title$2, null, "Member ID"), React__default.createElement(Value$1, null, holderId)), React__default.createElement(Section$1, null, React__default.createElement(Title$2, null, "Type"), React__default.createElement(Value$1, null, kind)), React__default.createElement(Section$1, null, React__default.createElement(Title$2, null, "Date"), React__default.createElement(Value$1, null, date)), React__default.createElement(Section$1, null, React__default.createElement(Title$2, null, "Address"), React__default.createElement(Value$1, null, address)), React__default.createElement(Section$1, null, React__default.createElement(Title$2, null, customTitle), React__default.createElement(Value$1, null, customValue)), React__default.createElement(Section$1, null, React__default.createElement(Title$2, null, "Schedule"), React__default.createElement(Row$2, null, startDate && React__default.createElement(Value$1, null, startDate), endDate && React__default.createElement(Value$1, null, "-"), endDate && React__default.createElement(Value$1, null, endDate))), React__default.createElement(Section$1, null, React__default.createElement(Title$2, null, "Offer description"), React__default.createElement(Value$1, null, description)), React__default.createElement(Section$1, {
    flexDirection: "row",
    alignItems: "center",
    gap: "12px"
  }, React__default.createElement(Svgs.PhoneIcon, {
    height: "16px",
    width: "16px"
  }), React__default.createElement(Title$2, null, "Commercial phone"), React__default.createElement(Value$1, null, phoneNumber)), React__default.createElement(Section$1, {
    flexDirection: "row",
    alignItems: "center",
    gap: "12px"
  }, React__default.createElement(Svgs.PhoneIcon, {
    height: "16px",
    width: "16px"
  }), React__default.createElement(Title$2, null, "After sale service"), React__default.createElement(Value$1, null, phoneNumberSecondary)), React__default.createElement(Section$1, {
    flexDirection: "row",
    alignItems: "center",
    gap: "12px"
  }, React__default.createElement(Svgs.GlobeIcon, {
    height: "16px",
    width: "16px"
  }), React__default.createElement(Value$1, null, website)), React__default.createElement(Section$1, {
    flexDirection: "row",
    alignItems: "center",
    gap: "12px"
  }, React__default.createElement(Svgs.GlobeIcon, {
    height: "16px",
    width: "16px"
  }), React__default.createElement(Value$1, null, disableNotifications)));
};

var _templateObject$c, _templateObject2$c;
var StyledDetailsBottonContainer = styled('div')(_templateObject$c || (_templateObject$c = _taggedTemplateLiteralLoose(["\n  .MuiButton-outlined {\n    border: 2px solid #d0d0d0;\n  }\n"])));
var StyledDetailsBotton = styled('button')(_templateObject2$c || (_templateObject2$c = _taggedTemplateLiteralLoose(["\n  font-family: 'Product Sans light', sans-serif;\n  margin: 0 auto;\n  font-weight: 600;\n  border-radius: 24px;\n  width: 100%;\n  margin: 15px 0 5px 0;\n  text-transform: capitalize;\n  padding: 8px;\n  display: flex;\n  align-items: center;\n  justify-content: center;\n  gap: 8px;\n  font-size: 14px;\n  font-weight: 500;\n  border: 1px solid #f2ecec;\n  background-color: #fff;\n  &:hover {\n    cursor: pointer;\n    background-color: #f5f5f5;\n  }\n  svg {\n    width: 12px;\n    height: 12px;\n    transition: 0.4s;\n    transform: ", ";\n    /* transform: rotate(-90deg); */\n  }\n"])), function (props) {
  return (!props.showDetails ? 'rotate3d(0,0,1,1turn)' : 'rotate3d(1,0,0,0.5turn)') + " rotate(-90deg)";
});

var DetailsButton = function DetailsButton(props) {
  var open = props.open,
    setOpen = props.setOpen;
  var onClick = function onClick() {
    setOpen(!open);
  };
  return React__default.createElement(StyledDetailsBottonContainer, null, React__default.createElement(StyledDetailsBotton, {
    showDetails: open,
    onClick: onClick
  }, React__default.createElement(Svgs.ExpandArrowIcon, null), React__default.createElement("span", null, "details")));
};

var _templateObject$d, _templateObject2$d, _templateObject3$a, _templateObject4$9;
var CardContainer$1 = styled("div")(_templateObject$d || (_templateObject$d = _taggedTemplateLiteralLoose(["\n  display: flex;\n  flex-direction: column;\n  width: min-content;\n  min-height: fit-content;\n  justify-content: center;\n  border-radius: 26px;\n  margin: 0 auto;\n  padding: 15px;\n  box-sizing: border-box;\n  /* Extra small devices (phones, 600px and down) */\n  /* @media only screen and (max-width: 600px) {\n    width: 100%;\n    border-radius: 0 10px 10px 10px;\n    padding: 16px 8px;\n    border: none;\n    margin: 0%;\n  } */\n"])));
var Card$1 = styled("div")(_templateObject2$d || (_templateObject2$d = _taggedTemplateLiteralLoose(["\n  display: flex;\n  flex-direction: column;\n  width: fit-content;\n  min-height: fit-content;\n  justify-content: center;\n  background: #fff;\n  border-radius: 26px;\n  margin: 0 auto;\n  box-sizing: border-box;\n  border: 1px solid #f2ecec;\n  overflow: hidden;\n  background-color: ", ";\n  min-width: 350px;\n  /* Extra small devices (phones, 600px and down) */\n  /* @media only screen and (max-width: 600px) {\n    width: 100%;\n    border-radius: 0 10px 10px 10px;\n    padding: 16px 8px;\n    border: none;\n    margin: 0%;\n  } */\n"])), function (_ref) {
  var backgroundColor = _ref.backgroundColor;
  return backgroundColor;
});
var CardContent = styled("div")(_templateObject3$a || (_templateObject3$a = _taggedTemplateLiteralLoose(["\n  padding: 14px 14px 0;\n"])));
var DetailsContainer = styled("div")(_templateObject4$9 || (_templateObject4$9 = _taggedTemplateLiteralLoose(["\n  display: flex;\n  flex-direction: column;\n  justify-content: flex-start;\n  align-items: flex-start;\n  margin-top: 10px;\n  height: ", ";\n  width: 100%;\n  box-sizing: border-box;\n  /* padding: ", "; */\n  padding: 0 14px;\n  opacity: ", ";\n  /* transition: height 0.3s cubic-bezier(0.4, 0, 1, 1),\n    opacity 1s cubic-bezier(0.4, 0, 1, 1); */\n  transition: ", ";\n  gap: 5px;\n"])), function (_ref2) {
  var open = _ref2.open;
  return open ? "auto" : "0";
}, function (_ref3) {
  var open = _ref3.open;
  return open ? "0px 10px 20px 10px" : "0";
}, function (_ref4) {
  var open = _ref4.open;
  return open ? "1" : "0";
}, function (_ref5) {
  var open = _ref5.open;
  return open ? "height 0.3s, opacity 0.5s cubic-bezier(0.4, 0, 1, 1);" : "height 0.3s";
});

var EventCardDetails$1 = function EventCardDetails(props) {
  var holderName = props.holderName,
    place = props.place,
    address = props.address,
    openTime = props.openTime,
    startDate = props.startDate,
    ticketNumber = props.ticketNumber,
    value = props.value,
    termsAndConditions = props.termsAndConditions,
    googleMapsLink = props.googleMapsLink,
    phoneNumber = props.phoneNumber,
    email = props.email,
    disableNotifications = props.disableNotifications;
  return React__default.createElement(Container$4, null, React__default.createElement(Section$1, null, React__default.createElement(Title$2, null, "Ticket holder"), React__default.createElement(Value$1, null, holderName)), React__default.createElement(Section$1, null, React__default.createElement(Title$2, null, place), React__default.createElement(Value$1, null, address)), React__default.createElement(Section$1, null, React__default.createElement(Title$2, null, "Date"), React__default.createElement(Value$1, null, openTime)), React__default.createElement(Section$1, null, React__default.createElement(Title$2, null, "Start date"), React__default.createElement(Value$1, null, startDate)), React__default.createElement(Section$1, null, React__default.createElement(Title$2, null, "Value"), React__default.createElement(Value$1, null, startDate)), React__default.createElement(Section$1, null, React__default.createElement(Title$2, null, "Ticket number"), React__default.createElement(Value$1, null, ticketNumber)), React__default.createElement(Section$1, null, React__default.createElement(Title$2, null, "Value"), React__default.createElement(Value$1, null, value)), React__default.createElement(Section$1, null, React__default.createElement(Title$2, null, "Terms and conditions"), React__default.createElement(Value$1, null, termsAndConditions)), React__default.createElement(Section$1, {
    flexDirection: "row",
    alignItems: "center",
    gap: "12px"
  }, React__default.createElement(Svgs.PositionIcon, {
    height: "16px",
    width: "16px"
  }), React__default.createElement(Value$1, null, googleMapsLink)), React__default.createElement(Section$1, {
    flexDirection: "row",
    alignItems: "center",
    gap: "12px"
  }, React__default.createElement(Svgs.PhoneIcon, {
    height: "16px",
    width: "16px"
  }), React__default.createElement(Value$1, null, phoneNumber)), React__default.createElement(Section$1, {
    flexDirection: "row",
    alignItems: "center",
    gap: "12px"
  }, React__default.createElement(Svgs.EmailIcon, {
    height: "16px",
    width: "16px"
  }), React__default.createElement(Value$1, null, email)), React__default.createElement(Section$1, {
    flexDirection: "row",
    alignItems: "center",
    gap: "12px"
  }, React__default.createElement(Svgs.GlobeIcon, {
    height: "16px",
    width: "16px"
  }), React__default.createElement(Value$1, null, disableNotifications)));
};

var _templateObject$e;
var Image$3 = styled('img')(_templateObject$e || (_templateObject$e = _taggedTemplateLiteralLoose(["\n  width: 100%;\n  height: 120px;\n  object-fit: cover;\n  box-shadow: rgb(0 0 0 / 16%) 0px 2px 3px 1px;\n"])));

var CouponCardDetails$1 = function CouponCardDetails(props) {
  var expires = props.expires,
    from = props.from,
    walletImage = props.walletImage,
    promoCode = props.promoCode,
    description = props.description,
    value = props.value,
    couponId = props.couponId,
    disableNotifications = props.disableNotifications;
  return React__default.createElement(Container$4, null, React__default.createElement(Section$1, null, React__default.createElement(Title$2, null, "Expires"), React__default.createElement(Value$1, null, expires)), React__default.createElement(Section$1, null, React__default.createElement(Title$2, null, "From"), React__default.createElement(Value$1, null, from), React__default.createElement(Image$3, {
    src: walletImage
  })), React__default.createElement(Section$1, null, React__default.createElement(Title$2, null, "Promo code"), React__default.createElement(Value$1, null, promoCode)), React__default.createElement(Section$1, null, React__default.createElement(Title$2, null, "Description"), React__default.createElement(Value$1, null, description)), React__default.createElement(Section$1, null, React__default.createElement(Title$2, null, "Value"), React__default.createElement(Value$1, null, value)), React__default.createElement(Section$1, null, React__default.createElement(Title$2, null, "ID coupon"), React__default.createElement(Value$1, null, couponId)), React__default.createElement(Section$1, {
    flexDirection: "row",
    alignItems: "center",
    gap: "12px"
  }, React__default.createElement(Svgs.GlobeIcon, {
    height: "16px",
    width: "16px"
  }), React__default.createElement(Value$1, null, disableNotifications)));
};

var _templateObject$f;
var Image$4 = styled('img')(_templateObject$f || (_templateObject$f = _taggedTemplateLiteralLoose(["\n  width: 100%;\n  height: 120px;\n  object-fit: cover;\n  box-shadow: rgb(0 0 0 / 16%) 0px 2px 3px 1px;\n"])));

var PlaneCardDetails$1 = function PlaneCardDetails(props) {
  var flightNumber = props.flightNumber,
    confirmationCode = props.confirmationCode,
    scheduledDepartureTime = props.scheduledDepartureTime,
    arrivalGate = props.arrivalGate,
    destinationTerminal = props.destinationTerminal,
    ticketImage = props.ticketImage,
    details = props.details,
    phoneNumber = props.phoneNumber,
    disableNotifications = props.disableNotifications;
  return React__default.createElement(Container$4, null, React__default.createElement(Section$1, null, React__default.createElement(Title$2, null, "Flight number"), React__default.createElement(Value$1, null, flightNumber)), React__default.createElement(Section$1, null, React__default.createElement(Title$2, null, "Confirmation code"), React__default.createElement(Value$1, null, confirmationCode)), React__default.createElement(Section$1, null, React__default.createElement(Title$2, null, "Scheduled departure time"), React__default.createElement(Value$1, null, scheduledDepartureTime)), React__default.createElement(Section$1, null, React__default.createElement(Title$2, null, "Arrival code"), React__default.createElement(Value$1, null, arrivalGate)), React__default.createElement(Section$1, null, React__default.createElement(Title$2, null, "Destination terminal"), React__default.createElement(Value$1, null, destinationTerminal), React__default.createElement(Image$4, {
    src: ticketImage
  })), React__default.createElement(Section$1, null, React__default.createElement(Title$2, null, "Flight details"), React__default.createElement(Value$1, null, details)), React__default.createElement(Section$1, {
    flexDirection: "row",
    alignItems: "center",
    gap: "12px"
  }, React__default.createElement(Svgs.PhoneIcon, {
    height: "16px",
    width: "16px"
  }), React__default.createElement(Value$1, null, phoneNumber)), React__default.createElement(Section$1, {
    flexDirection: "row",
    alignItems: "center",
    gap: "12px"
  }, React__default.createElement(Svgs.GlobeIcon, {
    height: "16px",
    width: "16px"
  }), React__default.createElement(Value$1, null, disableNotifications)));
};

var CinemaCardDetails$1 = function CinemaCardDetails(props) {
  var holderName = props.holderName,
    place = props.place,
    address = props.address,
    openTime = props.openTime,
    startDate = props.startDate,
    endDate = props.endDate,
    ticketNumber = props.ticketNumber,
    termsAndConditions = props.termsAndConditions,
    movieName = props.movieName,
    rating = props.rating,
    googleMapsLink = props.googleMapsLink,
    phoneNumber = props.phoneNumber,
    email = props.email;
  return React__default.createElement(Container$4, null, React__default.createElement(Section$1, null, React__default.createElement(Title$2, null, "Ticket holder"), React__default.createElement(Value$1, null, holderName)), React__default.createElement(Section$1, null, React__default.createElement(Title$2, null, place), React__default.createElement(Value$1, null, address)), React__default.createElement(Section$1, null, React__default.createElement(Title$2, null, "Doors open"), React__default.createElement(Value$1, null, openTime)), React__default.createElement(Section$1, null, React__default.createElement(Title$2, null, "Event start time"), React__default.createElement(Value$1, null, startDate)), React__default.createElement(Section$1, null, React__default.createElement(Title$2, null, "Event end time"), React__default.createElement(Value$1, null, endDate)), React__default.createElement(Section$1, null, React__default.createElement(Title$2, null, "Ticket number"), React__default.createElement(Value$1, null, ticketNumber)), React__default.createElement(Section$1, null, React__default.createElement(Title$2, null, "Terms and conditions"), React__default.createElement(Value$1, null, termsAndConditions)), React__default.createElement(Section$1, null, React__default.createElement(Title$2, null, "Movie name"), React__default.createElement(Value$1, null, movieName)), React__default.createElement(Section$1, null, React__default.createElement(Title$2, null, "Rating"), React__default.createElement(Value$1, null, rating)), React__default.createElement(Section$1, {
    flexDirection: "row",
    alignItems: "center",
    gap: "12px"
  }, React__default.createElement(Svgs.PositionIcon, {
    height: "16px",
    width: "16px"
  }), React__default.createElement(Value$1, null, googleMapsLink)), React__default.createElement(Section$1, {
    flexDirection: "row",
    alignItems: "center",
    gap: "12px"
  }, React__default.createElement(Svgs.PhoneIcon, {
    height: "16px",
    width: "16px"
  }), React__default.createElement(Value$1, null, phoneNumber)), React__default.createElement(Section$1, {
    flexDirection: "row",
    alignItems: "center",
    gap: "12px"
  }, React__default.createElement(Svgs.EmailIcon, {
    height: "16px",
    width: "16px"
  }), React__default.createElement(Value$1, null, email)));
};

var _templateObject$g;
var Image$5 = styled('img')(_templateObject$g || (_templateObject$g = _taggedTemplateLiteralLoose(["\n  width: 100%;\n  height: 120px;\n  object-fit: cover;\n  box-shadow: rgb(0 0 0 / 16%) 0px 2px 3px 1px;\n"])));

var BoatCardDetails$1 = function BoatCardDetails(props) {
  var startDate = props.startDate,
    endDate = props.endDate,
    price = props.price,
    number = props.number,
    ticketImage = props.ticketImage,
    details = props.details,
    googleMapsLink = props.googleMapsLink,
    phoneNumber = props.phoneNumber,
    disableNotifications = props.disableNotifications;
  return React__default.createElement(Container$4, null, React__default.createElement(Section$1, null, React__default.createElement(Title$2, null, "Departure"), React__default.createElement(Value$1, null, startDate)), React__default.createElement(Section$1, null, React__default.createElement(Title$2, null, "Arriving at"), React__default.createElement(Value$1, null, endDate)), React__default.createElement(Section$1, null, React__default.createElement(Title$2, null, "Price"), React__default.createElement(Value$1, null, price)), React__default.createElement(Section$1, null, React__default.createElement(Title$2, null, "Number"), React__default.createElement(Value$1, null, number), React__default.createElement(Image$5, {
    src: ticketImage
  })), React__default.createElement(Section$1, null, React__default.createElement(Title$2, null, "Details"), React__default.createElement(Value$1, null, details)), React__default.createElement(Section$1, {
    flexDirection: "row",
    alignItems: "center",
    gap: "12px"
  }, React__default.createElement(Svgs.PositionIcon, {
    height: "16px",
    width: "16px"
  }), React__default.createElement(Value$1, null, googleMapsLink)), React__default.createElement(Section$1, {
    flexDirection: "row",
    alignItems: "center",
    gap: "12px"
  }, React__default.createElement(Svgs.PhoneIcon, {
    height: "16px",
    width: "16px"
  }), React__default.createElement(Value$1, null, phoneNumber)), React__default.createElement(Section$1, {
    flexDirection: "row",
    alignItems: "center",
    gap: "12px"
  }, React__default.createElement(Svgs.GlobeIcon, {
    height: "16px",
    width: "16px"
  }), React__default.createElement(Value$1, null, disableNotifications)));
};

var _templateObject$h;
var Image$6 = styled('img')(_templateObject$h || (_templateObject$h = _taggedTemplateLiteralLoose(["\n  width: 100%;\n  height: 120px;\n  object-fit: cover;\n  box-shadow: rgb(0 0 0 / 16%) 0px 2px 3px 1px;\n"])));

var BusCardDetails = function BusCardDetails(props) {
  var startDate = props.startDate,
    endDate = props.endDate,
    price = props.price,
    number = props.number,
    ticketImage = props.ticketImage,
    details = props.details,
    googleMapsLink = props.googleMapsLink,
    phoneNumber = props.phoneNumber,
    disableNotifications = props.disableNotifications;
  return React__default.createElement(Container$4, null, React__default.createElement(Section$1, null, React__default.createElement(Title$2, null, "Departure"), React__default.createElement(Value$1, null, startDate)), React__default.createElement(Section$1, null, React__default.createElement(Title$2, null, "Arriving at"), React__default.createElement(Value$1, null, endDate)), React__default.createElement(Section$1, null, React__default.createElement(Title$2, null, "Price"), React__default.createElement(Value$1, null, price), React__default.createElement(Image$6, {
    src: ticketImage
  })), React__default.createElement(Section$1, null, React__default.createElement(Title$2, null, "Number"), React__default.createElement(Value$1, null, number)), React__default.createElement(Section$1, null, React__default.createElement(Title$2, null, "Details"), React__default.createElement(Value$1, null, details)), React__default.createElement(Section$1, {
    flexDirection: "row",
    alignItems: "center",
    gap: "12px"
  }, React__default.createElement(Svgs.PositionIcon, {
    height: "16px",
    width: "16px"
  }), React__default.createElement(Value$1, null, googleMapsLink)), React__default.createElement(Section$1, {
    flexDirection: "row",
    alignItems: "center",
    gap: "12px"
  }, React__default.createElement(Svgs.PhoneIcon, {
    height: "16px",
    width: "16px"
  }), React__default.createElement(Value$1, null, phoneNumber)), React__default.createElement(Section$1, {
    flexDirection: "row",
    alignItems: "center",
    gap: "12px"
  }, React__default.createElement(Svgs.GlobeIcon, {
    height: "16px",
    width: "16px"
  }), React__default.createElement(Value$1, null, disableNotifications)));
};

var _templateObject$i;
var Image$7 = styled('img')(_templateObject$i || (_templateObject$i = _taggedTemplateLiteralLoose(["\n  width: 100%;\n  height: 120px;\n  object-fit: cover;\n  box-shadow: rgb(0 0 0 / 16%) 0px 2px 3px 1px;\n"])));

var TrainCardDetails = function TrainCardDetails(props) {
  var startDate = props.startDate,
    endDate = props.endDate,
    price = props.price,
    number = props.number,
    ticketImage = props.ticketImage,
    details = props.details,
    googleMapsLink = props.googleMapsLink,
    phoneNumber = props.phoneNumber,
    disableNotifications = props.disableNotifications;
  return React__default.createElement(Container$4, null, React__default.createElement(Section$1, null, React__default.createElement(Title$2, null, "Departure"), React__default.createElement(Value$1, null, startDate)), React__default.createElement(Section$1, null, React__default.createElement(Title$2, null, "Arriving at"), React__default.createElement(Value$1, null, endDate)), React__default.createElement(Section$1, null, React__default.createElement(Title$2, null, "Price"), React__default.createElement(Value$1, null, price), React__default.createElement(Image$7, {
    src: ticketImage
  })), React__default.createElement(Section$1, null, React__default.createElement(Title$2, null, "Number"), React__default.createElement(Value$1, null, number)), React__default.createElement(Section$1, null, React__default.createElement(Title$2, null, "Transit details"), React__default.createElement(Value$1, null, details)), React__default.createElement(Section$1, {
    flexDirection: "row",
    alignItems: "center",
    gap: "12px"
  }, React__default.createElement(Svgs.PositionIcon, {
    height: "16px",
    width: "16px"
  }), React__default.createElement(Value$1, null, googleMapsLink)), React__default.createElement(Section$1, {
    flexDirection: "row",
    alignItems: "center",
    gap: "12px"
  }, React__default.createElement(Svgs.PhoneIcon, {
    height: "16px",
    width: "16px"
  }), React__default.createElement(Value$1, null, phoneNumber)), React__default.createElement(Section$1, {
    flexDirection: "row",
    alignItems: "center",
    gap: "12px"
  }, React__default.createElement(Svgs.GlobeIcon, {
    height: "16px",
    width: "16px"
  }), React__default.createElement(Value$1, null, disableNotifications)));
};

var RoomCardDetails$1 = function RoomCardDetails(props) {
  var holderName = props.holderName,
    place = props.place,
    address = props.address,
    termsAndConditions = props.termsAndConditions,
    kind = props.kind,
    startDate = props.startDate,
    endDate = props.endDate,
    googleMapsLink = props.googleMapsLink,
    phoneNumber = props.phoneNumber,
    email = props.email;
  return React__default.createElement(Container$4, null, React__default.createElement(Section$1, null, React__default.createElement(Title$2, null, "Ticket holder"), React__default.createElement(Value$1, null, holderName)), React__default.createElement(Section$1, null, React__default.createElement(Title$2, null, place), React__default.createElement(Value$1, null, address)), React__default.createElement(Section$1, null, React__default.createElement(Title$2, null, "Terms and conditions"), React__default.createElement(Value$1, null, termsAndConditions)), React__default.createElement(Section$1, null, React__default.createElement(Title$2, null, "Type"), React__default.createElement(Value$1, null, kind)), React__default.createElement(Section$1, null, React__default.createElement(Title$2, null, "Check in"), React__default.createElement(Value$1, null, startDate)), React__default.createElement(Section$1, null, React__default.createElement(Title$2, null, "Check out"), React__default.createElement(Value$1, null, endDate)), React__default.createElement(Section$1, {
    flexDirection: "row",
    alignItems: "center",
    gap: "12px"
  }, React__default.createElement(Svgs.PositionIcon, {
    height: "16px",
    width: "16px"
  }), React__default.createElement(Value$1, null, googleMapsLink)), React__default.createElement(Section$1, {
    flexDirection: "row",
    alignItems: "center",
    gap: "12px"
  }, React__default.createElement(Svgs.PhoneIcon, {
    height: "16px",
    width: "16px"
  }), React__default.createElement(Value$1, null, phoneNumber)), React__default.createElement(Section$1, {
    flexDirection: "row",
    alignItems: "center",
    gap: "12px"
  }, React__default.createElement(Svgs.EmailIcon, {
    height: "16px",
    width: "16px"
  }), React__default.createElement(Value$1, null, email)));
};

var CardAndroid = function CardAndroid(props) {
  var titleColor = props.titleColor,
    headerLogoLink = props.headerLogoLink,
    title = props.title,
    accessImage = props.accessImage,
    type = props.type,
    backgroundColor = props.backgroundColor;
  var _useState = React.useState(false),
    open = _useState[0],
    setOpen = _useState[1];
  var renderInfos = function renderInfos() {
    if (type === "boat" || type === "bus" || type === "subway" || type === "train") {
      return React__default.createElement(Infos, {
        topLeftTitle: props.topLeftTitle,
        topLeftValue: props.topLeftValue,
        topRightTitle: props.topRightTitle,
        topRightValue: props.topRightValue,
        bottomLeftTitle: props.bottomLeftTitle,
        bottomLeftValue: props.bottomLeftValue
      });
    }
    if (type === "event" || type === "cinema") {
      return React__default.createElement(Infos, {
        topLeftTitle: props.topLeftTitle,
        topLeftValue: props.topLeftValue,
        topRightTitle: props.topRightTitle,
        topRightValue: props.topRightValue,
        bottomLeftTitle: props.bottomLeftTitle,
        bottomLeftValue: props.bottomLeftValue,
        bottomCenterTitle: props.bottomCenterTitle,
        bottomCenterValue: props.bottomCenterValue,
        bottomRightTitle: props.bottomRightTitle,
        bottomRightValue: props.bottomRightValue
      });
    }
    if (type === "plane") {
      return React__default.createElement(Infos, {
        topLeftTitle: props.topLeftTitle,
        topLeftValue: props.topLeftValue,
        topCenterTitle: props.topCenterTitle,
        topCenterValue: props.topCenterValue,
        topRightTitle: props.topRightTitle,
        topRightValue: props.topRightValue,
        bottomLeftTitle: props.bottomLeftTitle,
        bottomLeftValue: props.bottomLeftValue,
        bottomRightTitle: props.bottomRightTitle,
        bottomRightValue: props.bottomRightValue
      });
    }
    if (type === "store") {
      return React__default.createElement(Infos, {
        topLeftTitle: props.topLeftTitle,
        topLeftValue: props.topLeftValue,
        topRightTitle: props.topRightTitle,
        topRightValue: props.topRightValue
      });
    }
    if (type === "room") {
      return React__default.createElement(Infos, {
        topLeftTitle: props.topLeftTitle,
        topLeftValue: props.topLeftValue,
        topCenterTitle: props.topCenterTitle,
        topCenterValue: props.topCenterValue,
        topRightTitle: props.topRightTitle,
        topRightValue: props.topRightValue
      });
    }
    if (type === "coupon") {
      return null;
    }
    return null;
  };
  var renderAccess = function renderAccess() {
    if (type === "plane") {
      return React__default.createElement(Access$1, {
        accessImage: accessImage,
        accessType: props.accessType,
        accessCodeId: props.accessCodeId
      });
    } else {
      return React__default.createElement(Access$1, {
        accessImage: accessImage,
        accessType: props.accessType,
        accessCodeId: props === null || props === void 0 ? void 0 : props.accessCodeId
      });
    }
  };
  var renderCardTitle = function renderCardTitle() {
    if (type === "room" || type === "event" || type === "cinema") {
      return React__default.createElement(CardTitle, {
        mainTitle: props.mainTitle,
        secondaryTitle: props.secondaryTitle,
        type: type
      });
    }
    if (type === "coupon" || type === "store") {
      return React__default.createElement(CardTitle, {
        mainTitle: props.mainTitle,
        type: type
      });
    }
    if (type === "boat" || type === "bus" || type === "subway" || type === "train" || type === "plane") {
      var _props$composedMainTi, _props$composedMainTi2, _props$composedSecond, _props$composedSecond2;
      return React__default.createElement(CardTitle, {
        composedMainTitle: {
          from: (_props$composedMainTi = props.composedMainTitle) === null || _props$composedMainTi === void 0 ? void 0 : _props$composedMainTi.from,
          to: (_props$composedMainTi2 = props.composedMainTitle) === null || _props$composedMainTi2 === void 0 ? void 0 : _props$composedMainTi2.to
        },
        composedSecondaryTitle: {
          from: (_props$composedSecond = props.composedSecondaryTitle) === null || _props$composedSecond === void 0 ? void 0 : _props$composedSecond.from,
          to: (_props$composedSecond2 = props.composedSecondaryTitle) === null || _props$composedSecond2 === void 0 ? void 0 : _props$composedSecond2.to
        },
        type: type
      });
    }
    return null;
  };
  var renderFooter = function renderFooter() {
    if (type !== "plane" && type !== "boat" && type !== "bus" && type !== "subway" && type !== "train") {
      return React__default.createElement(Footer, {
        bannerImage: props.bannerImage
      });
    }
    return null;
  };
  var renderCardDetails = function renderCardDetails() {
    if (type === "store") {
      return React__default.createElement(StoreCardDetails$1, {
        holderName: props.holderName,
        holderId: props.holderId,
        kind: props.kind,
        date: props.date,
        address: props.address,
        customTitle: props.customTitle,
        customValue: props.customValue,
        startDate: props.startDate,
        endDate: props.endDate,
        description: props.description,
        phoneNumber: props.phoneNumber,
        phoneNumberSecondary: props.phoneNumberSecondary,
        website: props.website,
        disableNotifications: props.disableNotifications
      });
    }
    if (type === "event") {
      return React__default.createElement(EventCardDetails$1, {
        holderName: props.holderName,
        place: props.place,
        address: props.address,
        openTime: props.openTime,
        startDate: props.startDate,
        ticketNumber: props.ticketNumber,
        value: props.value,
        termsAndConditions: props.termsAndConditions,
        googleMapsLink: props.googleMapsLink,
        phoneNumber: props.phoneNumber,
        email: props.email,
        disableNotifications: props.disableNotifications
      });
    }
    if (type === "cinema") {
      return React__default.createElement(CinemaCardDetails$1, {
        holderName: props.holderName,
        place: props.place,
        address: props.address,
        openTime: props.openTime,
        startDate: props.startDate,
        endDate: props.endDate,
        ticketNumber: props.ticketNumber,
        termsAndConditions: props.termsAndConditions,
        movieName: props.movieName,
        rating: props.rating,
        googleMapsLink: props.googleMapsLink,
        phoneNumber: props.phoneNumber,
        email: props.email
      });
    }
    if (type === "coupon") {
      return React__default.createElement(CouponCardDetails$1, {
        expires: props.expires,
        from: props.from,
        walletImage: props.walletImage,
        promoCode: props.promoCode,
        description: props.description,
        value: props.value,
        couponId: props.couponId,
        disableNotifications: props.disableNotifications
      });
    }
    if (type === "boat") {
      return React__default.createElement(BoatCardDetails$1, {
        startDate: props.startDate,
        endDate: props.endDate,
        price: props.price,
        number: props.number,
        ticketImage: props.ticketImage,
        details: props.details,
        googleMapsLink: props.googleMapsLink,
        phoneNumber: props.phoneNumber,
        disableNotifications: props.disableNotifications
      });
    }
    if (type === "bus") {
      return React__default.createElement(BusCardDetails, {
        startDate: props.startDate,
        endDate: props.endDate,
        price: props.price,
        number: props.number,
        ticketImage: props.ticketImage,
        details: props.details,
        googleMapsLink: props.googleMapsLink,
        phoneNumber: props.phoneNumber,
        disableNotifications: props.disableNotifications
      });
    }
    if (type === "subway") {
      return React__default.createElement(BusCardDetails, {
        startDate: props.startDate,
        endDate: props.endDate,
        price: props.price,
        number: props.number,
        ticketImage: props.ticketImage,
        details: props.details,
        googleMapsLink: props.googleMapsLink,
        phoneNumber: props.phoneNumber,
        disableNotifications: props.disableNotifications
      });
    }
    if (type === "train") {
      return React__default.createElement(TrainCardDetails, {
        startDate: props.startDate,
        endDate: props.endDate,
        price: props.price,
        number: props.number,
        ticketImage: props.ticketImage,
        details: props.details,
        googleMapsLink: props.googleMapsLink,
        phoneNumber: props.phoneNumber,
        disableNotifications: props.disableNotifications
      });
    }
    if (type === "room") {
      return React__default.createElement(RoomCardDetails$1, {
        holderName: props.holderName,
        place: props.place,
        address: props.address,
        termsAndConditions: props.termsAndConditions,
        kind: props.kind,
        startDate: props.startDate,
        endDate: props.endDate,
        email: props.email,
        googleMapsLink: props.googleMapsLink,
        phoneNumber: props.phoneNumber,
        disableNotifications: props.disableNotifications
      });
    }
    if (type === "plane") {
      return React__default.createElement(PlaneCardDetails$1, {
        flightNumber: props.flightNumber,
        confirmationCode: props.confirmationCode,
        scheduledDepartureTime: props.scheduledDepartureTime,
        arrivalGate: props.arrivalGate,
        destinationTerminal: props.destinationTerminal,
        ticketImage: props.ticketImage,
        details: props.details,
        phoneNumber: props.phoneNumber,
        disableNotifications: props.disableNotifications
      });
    } else return null;
  };
  return React__default.createElement(CardContainer$1, null, React__default.createElement(Card$1, {
    backgroundColor: backgroundColor
  }, React__default.createElement(Header$1, {
    titleColor: titleColor,
    headerLogoLink: headerLogoLink,
    title: title
  }), React__default.createElement(CardContent, null, renderCardTitle(), renderInfos(), renderAccess()), renderFooter()), React__default.createElement(DetailsButton, {
    open: open,
    setOpen: setOpen
  }), React__default.createElement(DetailsContainer, {
    open: open
  }, renderCardDetails()));
};

exports.CardAndroid = CardAndroid;
exports.CardIOS = CardIOS;
